---
title: "Orcas-Island-Time-Series-Metaproteomics"
author: "Miranda Mudge"
date: "2024-01-01"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```


```{r}
#packages needed

library(lubridate)
library(ggplot2)
library(tidyr)
library(MetBrewer)
library(tidyverse)
library(dplyr)
library(scales)
library(cowplot)
library(reshape2)
library(hms)
library(ggmap)
library(ggh4x)
library(rmarkdown)
library(readxl)
library(lattice)

```

## set working directory
```{r, setup, include=FALSE}
knitr::opts_knit$set(root.dir = 'D:/HAB/Biomarker_Manuscript/')
```

#########################################################################################################################
#########################################################################################################################
#########################################################################################################################

# Code for calculation and figures related to Figure 1

## color palettes
```{r}
library(MetBrewer)
met.sig20 <- met.brewer("Signac", 20) 

#set color pallet
met.sig14 <- met.brewer("Signac", 14) #14 colors
```

### Stacking all variables by Time

```{r}

combined_ts_2 <- read.csv("/HABTimeSeries_probedata.csv", header = T, na.strings = F)

# one way to make date into actual date
combined_ts_2 <- combined_ts_2 %>%
  mutate(Date = mdy(Date)) 

# convert time from character to time
combined_ts_2$Time <- as_hms(combined_ts_2$Time)

# make Date.Time into correct class
combined_ts_2$Date.Time = ymd_hms(paste(combined_ts_2$Date, combined_ts_2$Time))

# format dataframe to be tidy so you can plot
mm <- melt(subset(combined_ts_2, select=c(Date.Time, Chlorophyll.RFU, Conductivity.uscm, ODO.sat, Sal.psu, pH, Temp.C)), id.var="Date.Time")
```


Fix time in dataframe
- create a column for hour
- create a column for date and hour
```{r}
combined_ts_2$hour <- format(strptime(combined_ts_2$Time, "%H:%M"), "%H:00")

#Create a Date + Time column
combined_ts_2$Date.hour <- with(combined_ts_2, ymd(Date) + hm(hour)) #create column to group date.time by hour
```

Binning approach 
- find mean for every hour to smooth graph
```{r}
combined_bin1 <- combined_ts_2 %>% group_by(Date.hour) %>% 
  summarize(Chlorophyll.ugL = mean(Chlorophyll.ugL))    #group chl to match hour bins 

combined_bin2 <- combined_ts_2 %>% group_by(Date.hour) %>% 
  summarize(Temp.C = mean(Temp.C)) 

combined_bin3 <- combined_ts_2 %>% group_by(Date.hour) %>% 
  summarize(Conductivity.uscm = mean(Conductivity.uscm)) 

combined_bin4 <- combined_ts_2 %>% group_by(Date.hour) %>% 
  summarize(Sal.psu = mean(Sal.psu))

combined_bin5 <- combined_ts_2 %>% group_by(Date.hour) %>% 
  summarize(pH = mean(pH))

combined_bin6 <- combined_ts_2 %>% group_by(Date.hour) %>% 
  summarize(ODO.sat = mean(ODO.sat))
```

Combine all bins from above
```{r}
combined_bin <- merge(combined_bin1,combined_bin2,by="Date.hour")
combined_bin <- merge(combined_bin,combined_bin3,by="Date.hour")
combined_bin <- merge(combined_bin,combined_bin4,by="Date.hour")
combined_bin <- merge(combined_bin,combined_bin5,by="Date.hour")
combined_bin <- merge(combined_bin,combined_bin6,by="Date.hour")

combined_bin$Chlorophyll.ugL[combined_bin$Chlorophyll.ugL<0] <- 0
```

## Fixed ChlA plot - bin by hour
```{r}

#Chlorophyll vs time
a <- ggplot(combined_bin, aes(Date.hour, Chlorophyll.ugL))
ChlA.full <- a + geom_line(color = met.sig20[19]) + 
    geom_area(fill=met.sig20[19], alpha=0.8) + #, alpha=0.5
  ylim(-2, 100) +
  theme_bw() + 
  xlab("Date") +
  ylab(expression(bold(paste("Chlorophyll (",~mu,"g/L )", sep = "")))) + #adds micron symbol
  scale_x_datetime(labels = date_format("%d"), 
                   breaks = seq(from=as_datetime("2021-05-26"), by= "day", to=as_datetime("2021-06-19")),
                   expand = c(0, 0)) + #good line
  theme(text = element_text(size = 22, face="bold"),
    axis.text.x = element_text(vjust = 0.5, hjust=0.5),
        axis.text = element_text(color = "black"),
    axis.title.y  = element_text(face="bold"),
        panel.border = element_blank(), 
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(), 
        axis.line = element_line(colour = "black", size = 1.7),
    axis.ticks = element_line(colour = "black", size = 2)) #good line

```


```{r}
aaa <- ggplot(combined_bin, aes(Date.hour, Chlorophyll.ugL))
ChlA.discovery <- aaa + geom_line(color = met.sig20[19],size = 1.7) + 
    geom_area(fill=met.sig20[19], alpha=0.8) + #, alpha=0.5
  #ylim(-2, 100) +
  theme_bw() + 
  xlab("") +
  ylab(expression(bold(paste("Chlorophyll (",~mu,"g/L )", sep = "")))) + #adds micron symbol
  scale_x_datetime(labels = date_format("%d"), breaks = "24 hours",
                   limits = as.POSIXct(c('2021-06-12 00:00:00','2021-06-18 13:00:00')), 
                   expand = c(0,0)) + #good line
  scale_y_continuous(expand = c(0,0), limits = c(-2,100)) +
  theme(text = element_text(size = 22, face="bold"),
    axis.text.x = element_text(angle = 0, vjust = 0.5, hjust=0.5),
        axis.text = element_text(color = "black"),
    axis.title.y  = element_blank(),
        axis.title.x = element_blank(),
        panel.border = element_blank(), 
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(), 
        axis.line = element_line(colour = "black", size = 1.7),
    axis.ticks = element_line(colour = "black", size = 2)) #good line  

aaa <- ggplot(combined_bin, aes(Date.hour, Chlorophyll.ugL))
ChlA.validation <- aaa + geom_line(color = met.sig20[19],size = 1.7) + 
    geom_area(fill=met.sig20[19], alpha=0.8) + #, alpha=0.5
  #ylim(-2, 100) +
  theme_bw() + 
  xlab("") +
  ylab(expression(bold(paste("Chlorophyll (",~mu,"g/L )", sep = "")))) + #adds micron symbol
  scale_x_datetime(labels = date_format("%d"), breaks = "24 hours",
                   limits = as.POSIXct(c('2021-05-28 00:00:00','2021-06-02 24:00:00')), 
                   expand = c(0,0)) + #,+ #good line
  scale_y_continuous(expand = c(0,0), limits = c(-2,100)) +
  theme(text = element_text(size = 22, face="bold"),
    axis.text.x = element_text(angle = 0, vjust = 0.5, hjust=0.5),
        axis.text = element_text(color = "black"),
    axis.title.y  = element_blank(),
        axis.title.x = element_blank(),
        panel.border = element_blank(), 
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(), 
        axis.line = element_line(colour = "black", size = 1.7),
    axis.ticks = element_line(colour = "black", size = 2)) #good line  



#10x8
#gridExtra::grid.arrange(ChlA.validation, ncol=1)
```

save plot
```{r}
#Plot ChlA full plot

ggsave(file="/Desktop/Fig1_chlA_full.check.svg", plot=ChlA.full, width=10, height=5)
```

# BSD calcualtions

```{r}
ChlAflow <- read.csv("/Desktop/ProbeData.csv", header = T, na.strings = F)

#cols.num <- ChlAflow[,c(2:8)]
ChlAflow[,c(2:8)] <- sapply(ChlAflow[,c(2:8)],as.numeric)

ChlAflow$Date.ID <- strptime(paste0(ChlAflow$Date.hour), format = "%m/%d/%Y %H:%M")
ChlAflow$lubDate <- ymd_hms(ChlAflow$Date.ID)
ChlAflow$Chlorophyll.ugL[ChlAflow$Chlorophyll.ugL<0] <- 0
```

Differencing ChlA
```{r}

dY <- diff(ChlAflow$Chlorophyll.ugL)
dYdt <- diff(ChlAflow$Chlorophyll.ugL)/diff(ChlAflow$Time.passed)

#dX <- diff(phytoflow$Time.passed)  # the derivative of your function
dX <- rowMeans(embed(ChlAflow$Time.passed,2)) # centers the X values for plotting
#plot(ChlAflow$Time.passed[-1],dY,type="l",main="Derivative")

deriv.nano <- as.data.frame(cbind(dY, dX))
deriv.nano <- as.data.frame(cbind(deriv.nano, ChlAflow$Time.passed[-1]))
deriv.nano <- as.data.frame(cbind(deriv.nano, ChlAflow$Date.ID[-1]))
deriv.nano <- as.data.frame(cbind(deriv.nano, ChlAflow$Chlorophyll.ugL[-1]))
deriv.nano <- as.data.frame(cbind(deriv.nano, dYdt))


names(deriv.nano)[names(deriv.nano) == "ChlAflow$Date.ID[-1]"] <- "Date.ID"
names(deriv.nano)[names(deriv.nano) == "ChlAflow$Time.passed[-1]"] <- "Time.passed"
names(deriv.nano)[names(deriv.nano) == "ChlAflow$Chlorophyll.ugL[-1]"] <- "ChlA"

# grab end only
# last 36
discovery.bsd <- deriv.nano[c(351:484),]
validation.bsd <- deriv.nano[c(14:144),]

```

## Make plots for BSD validation
```{r}
bsd.chlA.validation <- ggplot(validation.bsd, aes(x = Date.ID)) + 
  geom_line(aes(y = dYdt), colour=met.sig20[19]) + 
  geom_vline(xintercept = as.POSIXct(as.POSIXct("0021-05-31 23:00:00")), 
             color="black",linetype = 2, size = 1) +
  ggtitle("validation Data - ChlA and BSD") +
    scale_x_datetime(labels = date_format("%m/%d"), breaks = "24 hours") + #,
  theme_bw() + 
  xlab("Date") +
  ylab(expression(bold(paste(Delta~"Chlorophyll (",~mu,"g/L ) /"~Delta~"time", sep = "")))) + 
    theme(text = element_text(size = 22, face="bold"),
    axis.text.x = element_text(angle = 45, vjust = 0.5, hjust=0.5),
        axis.text = element_text(color = "black"),
    axis.title.y  = element_text(face="bold"),
        panel.border = element_blank(), 
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(), 
        axis.line = element_line(colour = "black", size = 1.7),
    axis.ticks = element_line(colour = "black", size = 2))
```


```{r}
#Plot BSD validation

png(filename="/Desktop/Fig1_BSD_validation.png", 

   width = 500, height = 250,
    units = "px")

plot_grid( bsd.chlA.validation)

dev.off()
```


```{r}
#BSD discovery

 bsd.chlA.discovery <- ggplot(discovery.bsd, aes(x = Date.ID)) + 
  geom_line(aes(y = dYdt), colour=met.sig20[19]) + 
  geom_vline(xintercept = as.POSIXct(as.POSIXct("0021-06-18 11:00:00")), 
             color="black",linetype = 2, size = 1) +
  ggtitle("Discovery Data - ChlA and BSD") +
  scale_x_datetime(labels = date_format("%m/%d"), breaks = "24 hours") + #,
  theme_bw() + 
  xlab("Date") +
  ylab(expression(bold(paste(Delta~"Chlorophyll (",~mu,"g/L ) /"~Delta~"time", sep = "")))) + 
    theme(text = element_text(size = 22, face="bold"),
    axis.text.x = element_text(angle = 45, vjust = 0.5, hjust=0.5),
        axis.text = element_text(color = "black"),
    axis.title.y  = element_text(face="bold"),
        panel.border = element_blank(), 
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(), 
        axis.line = element_line(colour = "black", size = 1.7),
    axis.ticks = element_line(colour = "black", size = 2))
```

Plot BSD discovery bloom 
```{r}
#Plot BSD discovery

png(filename="/Desktop/Fig1_BSD_discovery.png", 

   width = 500, height = 250,
    units = "px")

plot_grid( bsd.chlA.discovery)

dev.off() 
```


## Flow cytometry
```{r}
# Read in files from excel spreadsheets
my_sheet_names <- excel_sheets("HAB_flowcytometry.xlsx")
my_sheets <- lapply(my_sheet_names, function(x) read_excel("HAB_flowcytometry.xlsx", sheet = x))
names(my_sheets) <- my_sheet_names

# Turn lists of excel sheets into dataframes
list2env(my_sheets, envir=.GlobalEnv)

# Combine Day and Hour into 1 column
Phytoplankton$DateID <- strptime(paste0(Phytoplankton$Date, Phytoplankton$Hour), format = "%Y-%m-%d %H")
Phytoplankton$lubDate <- ymd_hms(Phytoplankton$DateID)
Phytoplankton$DateID <- ymd_hms(Phytoplankton$lubDate)

# Bacteria dataframe
Bacteria$DateID <- strptime(paste0(Bacteria$Date, Bacteria$Hour), format = "%Y-%m-%d %H")
Bacteria$lubDate <- ymd_hms(Bacteria$DateID)
Bacteria$DateID <- ymd_hms(Bacteria$lubDate)

# Average Pikoeuks
Piko <- Phytoplankton %>% 
  group_by(DateID) %>% 
  summarise(AveragePikoeuks = mean(Picoeuks), StdevPikoeuks = sd(Picoeuks)) 


# Average Nanoeuks
Nanoeuks <- Phytoplankton %>% 
    select(Nanoeuks, Hour, DateID) %>%
  group_by(DateID) %>% 
  summarize(AverageNano = mean(Nanoeuks), Hour = mean(Hour), StdevNano = sd(Nanoeuks)) 

# Average Nanoeuks
Cyano <- Phytoplankton %>% 
    select(Cyanobacteria, Hour, DateID) %>%
  group_by(DateID) %>% 
  summarize(AverageCyano = mean(Cyanobacteria), Hour = mean(Hour), StdevNano = sd(Cyanobacteria)) 

#write.csv(Cyano, "/Desktop/cyanoData.csv")

# Average Bacteria
BactNumbers <- Bacteria %>% 
  group_by(DateID) %>% 
  summarise(AverageBacteria = mean(Bacteria), SDBacteria = sd(Bacteria))

phyto.merge <- merge(Nanoeuks, Piko, by = "DateID", all = T)
phyto.merge <- merge(phyto.merge, BactNumbers, by = "DateID", all = T)

#write.csv(phyto.merge, "/Desktop/PhytoAndFlowData_2.csv")

#write.csv(Bacteria, "/Desktop/bacterial_flowcytometry.csv")
```

Plot flow cytometry eukaryote cell counts
```{r}
flow.plot.discovery <- 
  ggplot(phyto.merge, aes(x = DateID)) + 
  geom_line(aes(y = AverageNano), color = met.sig14[3],  size = 1.7) + 
  geom_line(aes(y = AveragePikoeuks), color = met.sig14[11], size = 1.7) +
      geom_ribbon(aes(y = AverageNano, ymin = AverageNano - StdevNano, ymax = AverageNano + StdevNano), fill = met.sig14[3], alpha = .3) +
  geom_ribbon(aes(y = AveragePikoeuks, ymin = AveragePikoeuks - StdevPikoeuks, ymax = AveragePikoeuks + StdevPikoeuks), fill = met.sig14[11], alpha = .3) +
#ggtitle("Flow Cytometry Eukaryotes") +
  scale_x_datetime(labels = date_format("%d"), breaks = "24 hours",
                   limits = as.POSIXct(c('2021-06-12 00:00:00','2021-06-18 13:00:00')), 
                   expand = c(0,0)) + #,
    scale_y_continuous(expand = c(0,0), limits = c(-2,NA)) +
  theme_bw() + 
 # xlab("Date") +
  ylab("Cell counts (cells/mL)") + 
  theme(text = element_text(size = 22, face="bold"),
    axis.text.x = element_text(angle = 0, vjust = 0.5, hjust=0.5),
        axis.text = element_text(color = "black"),
    axis.title.y  = element_blank(),
        axis.title.x = element_blank(),
        panel.border = element_blank(), 
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(), 
        axis.line = element_line(colour = "black", size = 1.7),
    axis.ticks = element_line(colour = "black", size = 2)) 

flow.plot.validation <- 
  ggplot(phyto.merge, aes(x = DateID)) + 
  geom_line(aes(y = AverageNano), color = met.sig14[3],  size = 1.7) + 
  geom_line(aes(y = AveragePikoeuks), color = met.sig14[11], size = 1.7) +
      geom_ribbon(aes(y = AverageNano, ymin = AverageNano - StdevNano, ymax = AverageNano + StdevNano), fill = met.sig14[3], alpha = .3) +
  geom_ribbon(aes(y = AveragePikoeuks, ymin = AveragePikoeuks - StdevPikoeuks, ymax = AveragePikoeuks + StdevPikoeuks), fill = met.sig14[11], alpha = .3) +
#ggtitle("Flow Cytometry Eukaryotes") +
  scale_x_datetime(labels = date_format("%d"), breaks = "24 hours",
                   limits = as.POSIXct(c('2021-05-28 00:00:00','2021-06-02 24:00:00')), 
                   expand = c(0,0)) + #,
    scale_y_continuous(expand = c(0,0), limits = c(-2,NA)) +
  theme_bw() + 
 # xlab("Date") +
  ylab("Cell counts (cells/mL)") + 
  theme(text = element_text(size = 22, face="bold"),
    axis.text.x = element_text(angle = 0, vjust = 0.5, hjust=0.5),
        axis.text = element_text(color = "black"),
    axis.title.y  = element_blank(),
    axis.title.x = element_blank(),
        panel.border = element_blank(), 
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(), 
        axis.line = element_line(colour = "black", size = 1.7),
    axis.ticks = element_line(colour = "black", size = 2))

```

Save flow cytometry eukaryote plots
```{r}
#Plot Euk flow

png(filename="/Desktop/Fig1_flowEuks_discovery.png", 

   width = 500, height = 250,
    units = "px")

plot_grid( flow.plot.discovery)

dev.off() 

#################

png(filename="/Desktop/Fig1_flowEuks_validation.png", 

   width = 500, height = 250,
    units = "px")

plot_grid( flow.plot.validation)

dev.off() 
```

## Bacteria flow plot
```{r}
formatter1000 <- function(x){ 
    x/1000000 
}

flow.bact.discovery <- 
  ggplot(BactNumbers, aes(x = DateID)) + 
  geom_line(aes(y = AverageBacteria), colour=met.sig14[7], size = 1.7) + 
  geom_vline(xintercept = as.POSIXct(as.POSIXct("0021-06-18 11:00:00")), 
             color="black",linetype = 2, size = 1) +
  geom_ribbon(aes(y = AverageBacteria, ymin = AverageBacteria - SDBacteria, ymax = AverageBacteria + SDBacteria), fill =met.sig14[7], alpha = .3) +
  scale_x_datetime(labels = date_format("%d"), breaks = "24 hours",
                   limits = as.POSIXct(c('2021-06-12 00:00:00','2021-06-18 13:00:00')),
                   expand = c(0, 0)) + #,
    scale_y_continuous(limits = c(-2,6E6),labels = formatter1000, expand = c(0,0)) +
  theme_bw() + 
  xlab("") +
  ylab("Cell counts (cells/mL)") + 
  theme(text = element_text(size = 22, face="bold"),
    axis.text.x = element_text(angle = 0, vjust = 0.5, hjust=0.5),
        axis.text = element_text(color = "black"),
    axis.title.y  = element_blank(),
    axis.title.x = element_blank(),
        panel.border = element_blank(), 
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(), 
        axis.line = element_line(colour = "black", size = 1.7),
    axis.ticks = element_line(colour = "black", size = 2)) 
#+ylim(0,6E6)

flow.bact.validation <- 
  ggplot(phyto.merge, aes(x = DateID)) + 
  geom_line(aes(y = AverageBacteria), color = met.sig14[7], size = 1.7) + 
  geom_vline(xintercept = as.POSIXct(as.POSIXct("0021-06-18 11:00:00")), 
             color="black",linetype = 2, size = 1) +
  geom_ribbon(aes(y = AverageBacteria, ymin = AverageBacteria - SDBacteria, ymax = AverageBacteria + SDBacteria), fill = met.sig14[7], alpha = .3) +
 # ggtitle("Flow Cytometry Bacteria") +
  scale_x_datetime(labels = date_format("%d"), breaks = "24 hours",
                                      limits = as.POSIXct(c('2021-05-28 00:00:00','2021-06-02 24:00:00')), 
                   expand = c(0,0)) + #,
    scale_y_continuous(limits = c(-2, 6E6),labels = formatter1000, expand = c(0,0)) +
  theme_bw() + 
  xlab("") +
  ylab("Cell counts (cells/mL)") + 
  theme(text = element_text(size = 22, face="bold"),
    axis.text.x = element_text(angle = 0, vjust = 0.5, hjust=0.5),
        axis.text = element_text(color = "black"),
    axis.title.y  = element_blank(),
    axis.title.x = element_blank(),
        panel.border = element_blank(), 
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(), 
        axis.line = element_line(colour = "black", size = 1.7),
    axis.ticks = element_line(colour = "black", size = 2)) #+ylim(0,6E6)
```

Save bacteria plots
```{r}
#Plot Bacteria discovery

png(filename="/Desktop/Fig1_bactFlow_discovery.png", 

   width = 500, height = 250,
    units = "px")

plot_grid( flow.bact.discovery)

dev.off() 

#################

png(filename="/Desktop/Fig1_bactFlow_valiation.png", 

   width = 500, height = 250,
    units = "px")

plot_grid( flow.bact.validation)

dev.off() 
```

## Nutrients
```{r}
cov.dat <- read.csv("/HAB_2021_Nuts_metadata", header = T, na.strings = F)

cov.dat$DateTime2<-paste(cov.dat$Date, cov.dat$Time)
cov.dat$DateTime2<-parse_date_time(cov.dat$DateTime2, "mdy HM")

cov.nut<-subset(cov.dat, select=c('DateTime2', "NH4", 'PO4', 'Si.OH.4', 'NO3', 'NO2'))

nuts.discovery <- ggplot(cov.nut) +
    geom_line(aes(x=DateTime2, y=NH4), color=met.sig14[10], size=1.7) +   #purple
     geom_line(aes(x=DateTime2, y=PO4), color=met.sig14[8], size=1.7)+     #pink
    geom_line(aes(x=DateTime2, y=Si.OH.4), color=met.sig14[13], size=1.7)+  #teal green
    geom_line(aes(x=DateTime2, y=NO3), color=met.sig14[12], size=1.7)+       #coblt blue
    geom_line(aes(x=DateTime2, y=NO2), color=met.sig14[14], size=1.7)+      #lime green
  theme_minimal() + 
  xlab("Date") +
    ylab(expression(bold(paste("Concentration (mM)")))) + 
 # ggtitle("Nutrients discovery") +
  scale_x_datetime(labels = date_format("%d"), breaks = "24 hours",
                   limits = as.POSIXct(c('2021-06-12 00:00:00','2021-06-18 13:00:00')),
                   expand = c(0, 0)) + #note actually starts at 13:00
      scale_y_continuous(expand = c(0,0), limits = c(-2,NA)) +
  theme(text = element_text(size = 22, face="bold"),
    axis.text.x = element_text(angle = 0, vjust = 0.5, hjust=0.5),
        axis.text = element_text(color = "black"),
    axis.title.y  = element_blank(),
    axis.title.x = element_blank(),
        panel.border = element_blank(), 
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(), 
        axis.line = element_line(colour = "black", size = 1.7),
    axis.ticks = element_line(colour = "black", size = 2))
#nuts.discovery

nuts.validation <- ggplot(cov.nut) +
    geom_line(aes(x=DateTime2, y=NH4), color=met.sig14[10], size=1.7) +   #purple
     geom_line(aes(x=DateTime2, y=PO4), color=met.sig14[8], size=1.7)+     #pink
    geom_line(aes(x=DateTime2, y=Si.OH.4), color=met.sig14[13], size=1.7)+  #teal green
    geom_line(aes(x=DateTime2, y=NO3), color=met.sig14[12], size=1.7)+       #coblt blue
    geom_line(aes(x=DateTime2, y=NO2), color=met.sig14[14], size=1.7)+      #lime green
  theme_minimal() + 
  xlab("Date") +
    ylab(expression(bold(paste("Concentration (mM)")))) + 
 # ggtitle("Nutrients validation") +
  scale_x_datetime(labels = date_format("%d"), breaks = "24 hours",
                   limits = as.POSIXct(c('2021-05-28 00:00:00','2021-06-02 24:00:00')),
                   expand = c(0, 0)) + #,
      scale_y_continuous(expand = c(0,0), limits = c(-2,NA)) +
  theme(text = element_text(size = 22, face="bold"),
    axis.text.x = element_text(angle = 0, vjust = 0.5, hjust=0.5),
        axis.text = element_text(color = "black"),
    axis.title.y  = element_blank(),
    axis.title.x = element_blank(),
        panel.border = element_blank(), 
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(), 
        axis.line = element_line(colour = "black", size = 1.7),
    axis.ticks = element_line(colour = "black", size = 2))
  
#nuts.validation
```

Save nutrient plots
```{r}
#Plot nutrients 

png(filename="/Desktop/Fig1_nutrients_discovery.png", 

   width = 500, height = 250,
    units = "px")

plot_grid( nuts.discovery)

dev.off() 

#################

png(filename="/Desktop/Fig1_nutrients_validation.png", 

   width = 500, height = 250,
    units = "px")

plot_grid( nuts.validation)

dev.off() 
```

## Water Temperature 
```{r}
t.1 <- ggplot(combined_ts_2, aes(Date.Time))
probe.discovery <- t.1 + 
    geom_line(aes(y = Temp.C), color = met.sig14[4], size = 1.7) + 
   # geom_line(aes(y = Sal.psu),  color = Z[6], size = 1) + #out of range
  theme_minimal() + 
  xlab("June") +
    ylab(expression(bold(paste("")))) + 
#  ggtitle("Temp/Sal discovery") +
  scale_x_datetime(labels = date_format("%d"), breaks = "24 hours",
                   limits = as.POSIXct(c('2021-06-12 00:00:00','2021-06-18 13:00:00')),
                   expand = c(0, 0)) + #note actually starts at 13:00
        scale_y_continuous(expand = c(0,0), limits = c(8,20)) +
  theme(text = element_text(size = 22, face="bold"),
    axis.text.x = element_text(angle = 0, vjust = 0.5, hjust=0.5),
        axis.text = element_text(color = "black"),
    axis.title.y  = element_blank(),
    axis.title.x = element_text(size = 22, face="bold"),
        panel.border = element_blank(), 
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(), 
        axis.line = element_line(colour = "black", size = 1.7),
    axis.ticks = element_line(colour = "black", size = 2)) 


t.2 <- ggplot(combined_ts_2, aes(Date.Time))
probe.validation <- t.2 + 
    geom_line(aes(y = Temp.C), color = met.sig14[4], size = 1.7) + 
  #  geom_line(aes(y = Sal.psu), color = Z[6], size = 1) + #out of range
  theme_minimal() + 
  xlab("May - June") +
    ylab(expression(bold(paste("")))) + 
#  ggtitle("Temp/Sal validation") +
  scale_x_datetime(labels = date_format("%d"), breaks = "24 hours",
                   limits = as.POSIXct(c('2021-05-28 00:00:00','2021-06-02 24:00:00')), 
                   expand = c(0,0)) + #,
        scale_y_continuous(expand = c(0,0), limits = c(8,20)) +
  theme(text = element_text(size = 22, face="bold"),
    axis.text.x = element_text(angle = 0, vjust = 0.5, hjust=0.5),
        axis.text = element_text(color = "black"),
    axis.title.y  = element_blank(),
    axis.title.x = element_text(size = 22, face="bold"),
        panel.border = element_blank(), 
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(), 
        axis.line = element_line(colour = "black", size = 1.7),
    axis.ticks = element_line(colour = "black", size = 2))
  
```

Save temperature plots
```{r}
#Plot temp 

png(filename="/Desktop/Fig1_TempSal_discovery.png", 

   width = 500, height = 250,
    units = "px")

plot_grid(probe.discovery)

dev.off() 

#################

png(filename="/Desktop/Fig1_TempSal_validation.png", 

   width = 500, height = 250,
    units = "px")

plot_grid(probe.validation)

dev.off() 
```

################

Make Figure 1
```{r}
gA <- ggplotGrob(flow.plot.validation)
gB <- ggplotGrob(flow.bact.validation)
gC <- ggplotGrob(nuts.validation)
gD <- ggplotGrob(probe.validation)

gF <- ggplotGrob(flow.plot.discovery)
gG <- ggplotGrob(flow.bact.discovery)
gH <- ggplotGrob(nuts.discovery)
gI <- ggplotGrob(probe.discovery)

gK <- ggplotGrob(ChlA.discovery)
gL <- ggplotGrob(ChlA.validation)

```

Save Discovery bloom subplots
```{r}

png(filename="/Desktop/Fig1_all_discovery.png", 

   width = 300, height = 1000,
    units = "px")

grid::grid.newpage()
grid::grid.draw(rbind(gK, gF, gG, gH, gI))

dev.off() 

#################
```

Save Validation bloom subplots
```{r}

png(filename="/Desktop/Fig1_all_validation.png", 

   width = 300, height = 1000,
    units = "px")

grid::grid.newpage()
grid::grid.draw(rbind(gL, gA, gB, gC, gD))

dev.off() 

#################
```


#######################################
#######################################
#######################################

# Supplemental Figure 1

## Salinity
```{r}
sal.1 <- ggplot(combined_ts_2, aes(Date.Time))
sal.discovery <- t.1 + 
    geom_line(aes(y = Sal.psu),  color = met.sig14[3], size = 1.7) + #out of range
  theme_minimal() + 
  xlab("June") +
  ylab("Salinity (psu)") +
  scale_x_datetime(labels = date_format("%d"), breaks = "24 hours",
                   limits = as.POSIXct(c('2021-06-12 00:00:00','2021-06-18 13:00:00')),
                   expand = c(0, 0)) + #note actually starts at 13:00
        scale_y_continuous(expand = c(0,0), limits = c(29, 35.0)) +
  theme(text = element_text(size = 22, face="bold"),
    axis.text.x = element_text(angle = 0, vjust = 0.5, hjust=0.5),
        axis.text = element_text(color = "black"),
    axis.title.y  = element_text(size = 22, face="bold"),
    axis.title.x = element_blank(),
        panel.border = element_blank(), 
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(), 
        axis.line = element_line(colour = "black", size = 1.7),
    axis.ticks = element_line(colour = "black", size = 2)) 
  
#sal.discovery

sal.2 <- ggplot(combined_ts_2, aes(Date.Time))
sal.validation <- t.2 + 
    geom_line(aes(y = Sal.psu),  color = met.sig14[3], size = 1.7) + #out of range
  theme_minimal() + 
  xlab("May - June") +
  ylab("Salinity (psu)") +
  scale_x_datetime(labels = date_format("%d"), breaks = "24 hours",
                   limits = as.POSIXct(c('2021-05-28 00:00:00','2021-06-02 24:00:00')), 
                   expand = c(0,0)) + #,
        scale_y_continuous(expand = c(0,0), limits = c(29, 35.0)) +
  theme(text = element_text(size = 22, face="bold"),
    axis.text.x = element_text(angle = 0, vjust = 0.5, hjust=0.5),
        axis.text = element_text(color = "black"),
    axis.title.y  = element_text(size = 22, face="bold"),
    axis.title.x = element_blank(),
        panel.border = element_blank(), 
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(), 
        axis.line = element_line(colour = "black", size = 1.7),
    axis.ticks = element_line(colour = "black", size = 2))
  
#sal.validation
```

## pH
```{r}
pH.1 <- ggplot(combined_ts_2, aes(Date.Time))
pH.discovery <- t.1 + 
    geom_line(aes(y = pH),  color = met.sig14[2], size = 1.7) + #out of range
  theme_minimal() + 
  xlab("June") +
  ylab("pH") +
  scale_x_datetime(labels = date_format("%d"), breaks = "24 hours",
                   limits = as.POSIXct(c('2021-06-12 00:00:00','2021-06-18 13:00:00')),
                   expand = c(0, 0)) + #note actually starts at 13:00
        scale_y_continuous(expand = c(0,0), limits = c(7.5, 8.6)) +
  theme(text = element_text(size = 22, face="bold"),
    axis.text.x = element_text(angle = 0, vjust = 0.5, hjust=0.5),
        axis.text = element_text(color = "black"),
    axis.title.y  = element_text(size = 22, face="bold"),
    axis.title.x = element_blank(),
        panel.border = element_blank(), 
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(), 
        axis.line = element_line(colour = "black", size = 1.7),
    axis.ticks = element_line(colour = "black", size = 2)) 
  
#pH.discovery

ph.2 <- ggplot(combined_ts_2, aes(Date.Time))
pH.validation <- t.2 + 
    geom_line(aes(y = pH),  color = met.sig14[2], size = 1.7) + #out of range
  theme_minimal() + 
  xlab("May - June") +
  ylab("pH") +
  scale_x_datetime(labels = date_format("%d"), breaks = "24 hours",
                   limits = as.POSIXct(c('2021-05-28 00:00:00','2021-06-02 24:00:00')), 
                   expand = c(0,0)) + #,
        scale_y_continuous(expand = c(0,0), limits = c(7.5, 8.6)) +
  theme(text = element_text(size = 22, face="bold"),
    axis.text.x = element_text(angle = 0, vjust = 0.5, hjust=0.5),
        axis.text = element_text(color = "black"),
    axis.title.y  = element_text(size = 22, face="bold"),
    axis.title.x = element_blank(),
        panel.border = element_blank(), 
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(), 
        axis.line = element_line(colour = "black", size = 1.7),
    axis.ticks = element_line(colour = "black", size = 2))
  
#pH.validation
```

## Dissolved oxygen
```{r}
ODO.1 <- ggplot(combined_ts_2, aes(Date.Time))
ODO.discovery <- t.1 + 
    geom_line(aes(y = ODO.sat),  color = met.sig14[9], size = 1.7) + #out of range
  theme_minimal() + 
  xlab("June") +
  ylab("ODO (% saturation)") +
  scale_x_datetime(labels = date_format("%d"), breaks = "24 hours",
                   limits = as.POSIXct(c('2021-06-12 00:00:00','2021-06-18 13:00:00')),
                   expand = c(0, 0)) + #note actually starts at 13:00
        scale_y_continuous(expand = c(0,0), limits = c(55, 160)) +
  theme(text = element_text(size = 22, face="bold"),
    axis.text.x = element_text(angle = 0, vjust = 0.5, hjust=0.5),
        axis.text = element_text(color = "black"),
    axis.title.y  = element_text(size = 22, face="bold"),
    axis.title.x = element_text(size = 22, face="bold"),
        panel.border = element_blank(), 
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(), 
        axis.line = element_line(colour = "black", size = 1.7),
    axis.ticks = element_line(colour = "black", size = 2)) 
  
#ODO.discovery

ODO.2 <- ggplot(combined_ts_2, aes(Date.Time))
ODO.validation <- t.2 + 
    geom_line(aes(y = ODO.sat),  color = met.sig14[9], size = 1.7) + #out of range
  theme_minimal() + 
  xlab("May - June") +
  ylab("ODO (% saturation)") +
  scale_x_datetime(labels = date_format("%d"), breaks = "24 hours",
                   limits = as.POSIXct(c('2021-05-28 00:00:00','2021-06-02 24:00:00')), 
                   expand = c(0,0)) + #,
        scale_y_continuous(expand = c(0,0), limits = c(55, 160)) +
  theme(text = element_text(size = 22, face="bold"),
    axis.text.x = element_text(angle = 0, vjust = 0.5, hjust=0.5),
        axis.text = element_text(color = "black"),
    axis.title.y  = element_text(size = 22, face="bold"),
    axis.title.x = element_text(size = 22, face="bold"),
        panel.border = element_blank(), 
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(), 
        axis.line = element_line(colour = "black", size = 1.7),
    axis.ticks = element_line(colour = "black", size = 2))
  
#ODO.validation
```

```{r}
gM <- ggplotGrob(sal.discovery)
gN <- ggplotGrob(pH.discovery)
gP <- ggplotGrob(ODO.discovery)

gQ <- ggplotGrob(sal.validation)
gR <- ggplotGrob(pH.validation)
gS <- ggplotGrob(ODO.validation)

Supp_Fig1_validation_all <- rbind(gM, gN, gP)
Supp_Fig1_discovery_all <- rbind(gQ, gR, gS)

```

Save Discovery bloom subplots
```{r}

png(filename="/Desktop/Supp_Fig1_all_discovery.png", 

   width = 300, height = 700,
    units = "px")

grid::grid.newpage()
grid::grid.draw(rbind(gM, gN, gP))

dev.off() 

#################
```

Save Validation bloom subplots
```{r}

png(filename="/Desktop/Supp_Fig1_all_validation.png", 

   width = 300, height = 700,
    units = "px")

grid::grid.newpage()
grid::grid.draw(rbind(gQ, gR, gS))

dev.off() 

#################
```


#########################################################################################################################
#########################################################################################################################
#########################################################################################################################

# No code for Figure 2

#########################################################################################################################
#########################################################################################################################
#########################################################################################################################

# Code for calculation and figures related to Figure 3

# Phase 1 A
## Data dimensionality reduction
Code for calculation and figures related to Figure 3A. 
- code provided by Jea Park, amended by MCM
```{r Fig 3A}
require(readr)
require(ggplot2)
require(tidyr)
require(dplyr)
require(plotly)
require(WGCNA)
require(flashClust)

# median location normalization 
normalize.median=function (df){
  df_med = apply(as.matrix(df), 2, function(x) median(x, na.rm = T))
  df_med_loc = df_med - median(df_med)
  df.median = t(t(df)-df_med_loc)
  return(df.median)
}

# read in data
df <- read_csv("D:/HAB/2022_09_19_HAB_DDA_metagenome_corrected/HAB_2021_DDA_metagenome_contam_v1.4.10-1.12.31_TICnormalized_092722_36samples_final_Rcleaned.csv")
meta.dt <- read_csv("D:/HAB_WGCNA_test/HAB_2021_WGCNA_datTraits_corr1.csv")

# clean dataframe
rnames = as.matrix(df[,1])
df = df[,-1]
df[df == 0] = NA #all zeros into NA
df.temp = as.data.frame(df)
rownames(df) <- rnames

## log2 transform ##############################################################

df.log = log2(df)
df.median = normalize.median(df.log)
rownames(df.median) <- rownames(df)


## PCA Analysis ################################################################

data.matrix = df.median

data.matrix[is.na(data.matrix)] = 0

res <- svd(data.matrix-rowMeans(data.matrix))

pca <- tryCatch({
  prcomp(t(data.matrix), retx = TRUE, center = TRUE, scale = TRUE)
}, error = function(err) {
  prcomp(t(data.matrix), retx = TRUE, center = TRUE, scale = FALSE)
})
x=1
y=2
z=3
pcVar <- round((res$d^2)/sum(res$d^2) * 100, 2)

xlab <- sprintf(paste0("PC",x,": %.2f%% var"), pcVar[x])
ylab <- sprintf(paste0("PC",y,": %.2f%% var"), pcVar[y])
zlab <- sprintf(paste0("PC",z,": %.2f%% var"), pcVar[z])

# set color palette
met.sig20 <- met.brewer("Signac", 20) 

# create plot of PC2 v PC3
pc <- pca$x
pc = cbind(as.data.frame(pc),batch=as.factor(meta.dt$Date)) ###

fig <-plot_ly(pc, x = ~PC2, y = ~PC3,  color = ~batch, 
               text= ~paste0("Day ", batch,":",rownames(pc)), hoverinfo="text",  
               marker = list(size = 15),
               colors = c( met.sig20[11],met.sig20[11], met.sig20[11], met.sig20[11],
                           met.sig20[16], met.sig20[16] ,  met.sig20[16])) ###
fig <- fig %>% add_markers(size=1.5)
fig <- fig %>% layout(#title = 'Pre-adjustment PCA', ###
                      scene = list(xaxis = list(title = xlab, showgrid = FALSE, color = "black"),
                                   yaxis = list(title = ylab, showgrid = FALSE, color = "black")),
                      xaxis = list(range=c(-70,70),tickfont = list(size = 20, color = "black"),
                                   titlefont = list(size = 30, color = "black", size = 2),
                                   title = "PC2", showgrid = FALSE, color = "black"), 
                      yaxis = list(range=c(-70,70),tickfont = list(size = 20, color = "black"),
                                   titlefont = list(size = 30, color = "black", size = 2),
                                   title = "PC3", showgrid = FALSE),
                      showlegend = F)

Fig_pca <- fig

# manual export 800 x 800
#Fig3A_new_pinkpurple_800x800.png
#supp_fig_PCA12

a <- pca$rotation

Phase1A <- a %>% as.data.frame %>% rownames_to_column %>% 
  dplyr::select(rowname, PC2) %>% arrange(desc(abs(PC2))) %>% head(447)


# write file for top loading peptides for PC2
# write.csv(Phase1A, "D:/HAB/PCA_peptideLoadingsPC2_desc447.csv")
```

Mann - Whitney test on PC time series separation
```{r}
summary(pca)
# Proportion of variance PC1 = 12.67%
# Proportion of variance PC2 = 9.17%
# Proportion of variance PC3 = 6.67%

#pc_samples <- as.data.frame(t(pc))

# separate first and second half of time series by PC variance (16.17 - 19.21 vs 20.01 - 22.13)
test_pc1_1 <- pc$PC1[1:20]
test_pc1_2 <- pc$PC1[21:36]
# W = 171, p-value = 0.7413

test_pc2_1 <- pc$PC2[1:20]
test_pc2_2 <- pc$PC2[21:36]
# W = 317, p-value = 1.916e-09

test_pc3_1 <- pc$PC3[1:20]
test_pc3_2 <- pc$PC3[21:36]
# W = 42, p-value = 7.046e-05

wilcox.test(test_pc1_1,test_pc1_2)
wilcox.test(test_pc2_1,test_pc2_2)
wilcox.test(test_pc3_1,test_pc3_2)

```


# Supplemental figure 5
- PC1 v PC2
```{r Fig 3A}
# create plot of PC1 v PC2
pc <- pca$x
pc = cbind(as.data.frame(pc),batch=as.factor(meta.dt$Date)) ###

fig_supp <-plot_ly(pc, x = ~PC1, y = ~PC2,  color = ~batch, 
               text= ~paste0("Day ", batch,":",rownames(pc)), hoverinfo="text",  
               marker = list(size = 15),
               colors = c( met.sig20[11],met.sig20[11], met.sig20[11], met.sig20[11],
                           met.sig20[16], met.sig20[16] ,  met.sig20[16])) ###
fig_supp <- fig_supp %>% add_markers(size=1.5)
fig_supp <- fig_supp %>% layout(#title = 'Pre-adjustment PCA', ###
                      scene = list(xaxis = list(title = xlab, showgrid = FALSE, color = "black"),
                                   yaxis = list(title = ylab, showgrid = FALSE, color = "black")),
                      xaxis = list(range=c(-70,70),tickfont = list(size = 20, color = "black"),
                                   titlefont = list(size = 30, color = "black", size = 2),
                                   title = "PC1", showgrid = FALSE, color = "black"), 
                      yaxis = list(range=c(-70,70),tickfont = list(size = 20, color = "black"),
                                   titlefont = list(size = 30, color = "black", size = 2),
                                   title = "PC2", showgrid = FALSE),
                      showlegend = F)

Fig_pca_supp <- fig_supp

# manual export 800 x 800
#supp_fig_PCA12.png

```

######################################################################################################
######################################################################################################
######################################################################################################

# Phase 1 B
## Taxonomic correlation analysis

Code for calculation and figures related to Figure 3A.
- code by MCM

Read in file
```{r}
#read in taxonomic data generated by MetaGOmics analysis

taxa_counts <-  read.csv("D:/HAB/2022_09_19_HAB_DDA_metagenome_corrected/MetaGOmics/HAB_2021_taxa_counts_annotations_all.csv", header = T, na.strings = F)

```

Wrangle data
```{r}
taxa_Run_ratio <- taxa_counts

# clean names
names(taxa_Run_ratio)[names(taxa_Run_ratio) == "GO acc"] <- "GO.acc"
names(taxa_Run_ratio)[names(taxa_Run_ratio) == "GO aspect"] <- "GOtype"
names(taxa_Run_ratio)[names(taxa_Run_ratio) == "GO name"] <- "GOterm"
names(taxa_Run_ratio)[names(taxa_Run_ratio) == "taxon name"] <- "taxon"
names(taxa_Run_ratio)[names(taxa_Run_ratio) == "taxonomy rank"] <- "taxonrank"

#select only class
taxa_class <- taxa_Run_ratio
taxa_class <- taxa_class[taxa_class['taxonrank'] == "class", ]

#make dataframe numeric
taxa_class <- taxa_class[!duplicated(taxa_class),] 
taxa_test <- lapply(taxa_class[7:42],as.numeric)
taxa_test <- as.data.frame(taxa_test)
taxa_class_test <- cbind(taxa_class[1:5], taxa_test)
taxa_class <- taxa_class_test

#grab only BP
taxa_class_bp <- taxa_class[taxa_class['GOterm'] == "biological_process", ]
taxa_class_bp <- taxa_class_bp[c(5,6:41)] # counts

```

Set color palette 
Set label formatter (report peptide abundance / 1e6)
```{r}
library(MetBrewer)

met_rtb <- met.brewer("Troy", 7) #red to blue, good for correlaton plot


formatter1000 <- function(x){ 
    x/1000000 
}
```

Supplemental Figure 6
Summed peptide abundance over time by bacterial class
```{r}
# calculate maximum peptide abundance per class - used to order classes by abundance
taxa_class_bp$max <- apply(taxa_class_bp[2:27], 1, max, na.rm=TRUE)
# make data tidy
total_ratios <- gather(taxa_class_bp, Sample, Ratio, c(2:37), factor_key=TRUE)

# create column for ordering data by class
total_ratios$facet = factor(total_ratios$taxon, levels = c("Alphaproteobacteria",
"Gammaproteobacteria",
"Flavobacteriia",
"Verrucomicrobiae",
"Bacilli",
"Opitutae",
"Betaproteobacteria",
"Clostridia",
"Cytophagia",
"Gloeobacteria",
"Deltaproteobacteria",
"Epsilonproteobacteria",
"Planctomycetia",
"Lentisphaeria",
"Sphingobacteriia",
"Chlamydiia",
"Zetaproteobacteria",
"Saprospiria",
"Bacteroidia",
"Phycisphaerae",
"Acidobacteriia",
"Tissierellia",
"Acidimicrobiia",
"Oligoflexia",
"Chitinophagia",
"Actinobacteria",
"Spirochaetia",
"Thermotogae",
"Balneolia"
))

# amend sample names to only show time point ##.## (Day.Hour)
total_ratios$Sample <- substr(total_ratios$Sample, 10, 14)

# plot
total_plot <- ggplot(total_ratios, aes(Sample, Ratio, group=1))
total_plot <- total_plot +
  geom_point(size = 4) +
  theme_bw() + 
  facet_wrap(~facet, scales='free_y')+
    theme(legend.position="none",
          text = element_text(size = 40, color = "black"),
          axis.text.x = element_text( vjust = 0.5, hjust=0.5,color = "black", size =40),
          axis.text.y = element_text(color = "black", size =40),
          axis.title.x = element_text(vjust=-1, size = 40, color = "black"),
          axis.title.y  = element_text(color = "black",size = 40),
          panel.border = element_blank(), 
          panel.grid.major = element_blank(),
          panel.grid.minor = element_blank(), 
          axis.line = element_line(colour = "black", size = 2),
          axis.ticks = element_line(colour = "black", size = 2),
          panel.background = element_blank(),
          strip.background = element_blank(),
          strip.text = element_text(size = 35, colour = "black")) +
  scale_x_discrete(breaks=c("17.01","18.01", "19.01", 
                            "20.01", "21.01", "22.01"),
        labels=c("13","14", "15", "16", 
                 "17", "18"))  +
  scale_y_continuous(limits = c(0, NA),labels = formatter1000) +
  ylab("Peptide Abundance (x10^6)") + 
  xlab("June") 

total_plot

# manual export
#Supp_Fig_6_3700x2000.png
```

Create subset of classes with positive correlation over time
```{r}
# name
taxon <- c(
"Verrucomicrobiae",
"Opitutae",
"Zetaproteobacteria"
)

# color
    new <- c(
met_rtb[5],#0.9
met_rtb[5], #0.85
 met_rtb[5] #0.64
)

# bind name and color
color_bigplot_pos <- as.data.frame(cbind(taxon, new))

# add summed peptide abundances to dataframe
total_ratios_pos <- merge(total_ratios, color_bigplot_pos, by = "taxon", all.y = T)
```

Create subset of classes with negative correlation over time
```{r}
# name
taxon <- c(
  "Flavobacteriia",
"Gammaproteobacteria",
"Epsilonproteobacteria"
)

# color    
    new <- c(
 met_rtb[4], #-0.54
 met_rtb[4], #-0.56
 met_rtb[4] #-0.55
)

# bind name and color
color_bigplot_neg <- as.data.frame(cbind(taxon, new))

# add summed peptide abundances to dataframe
total_ratios_neg <- merge(total_ratios, color_bigplot_neg, by = "taxon", all.y = T)
```

Create subset of classes with no correlation over time
```{r}
# name
taxon <- c(
  "Alphaproteobacteria", 
"Bacilli",
"Betaproteobacteria"
)

# color     
    new<- c("black" , #0.35
"black", #-0.23
 "black" #0
)

# bind name and color
color_bigplot_neut <- as.data.frame(cbind(taxon, new))

# add summed peptide abundances to dataframe
total_ratios_neut <- merge(total_ratios, color_bigplot_neut, by = "taxon", all.y = T)
```

Combine classes for taxonomic correlation analysis figure
```{r}
# bind correlation dfs
total_ratios_pnn <- rbind(total_ratios_pos, total_ratios_neg, total_ratios_neut)

# order columns
total_ratios_pnn$order = factor(total_ratios_pnn$taxon, levels = c(

"Verrucomicrobiae",
"Flavobacteriia",
"Alphaproteobacteria", 

"Opitutae",
"Gammaproteobacteria",
"Bacilli",

"Zetaproteobacteria",
"Epsilonproteobacteria",
"Betaproteobacteria"
))

# amend sample names to only show time point ##.## (Day.Hour)
#total_ratios_pnn$Sample <- substr(total_ratios_pnn$Sample, 10, 14)

```

Plot for Fig 3B
```{r}
# Set colors based on the levels of 'new'
colors_pnn <- c("#7ba0b4", "#c27668", "black") 

#class level analysis
total_plot_fig <- ggplot(total_ratios_pnn, aes(factor(Sample), Ratio, group=Sample, color = new))
total_plot_fig <- total_plot_fig +
  geom_point( show.legend=F, color = total_ratios_pnn$new, size = 2.7) +
  geom_line(aes(group = 1), stat='smooth', method = "loess", alpha=0.1) +
    geom_ribbon( aes(group=1, fill = new), 
                stat='smooth', method = "loess", se=TRUE, alpha = 0.4,colour = NA) +
    scale_color_manual(values = colors_pnn) +  # Set manual colors for geom_smooth
  scale_fill_manual(values = colors_pnn) +
  theme_bw() + 
  facet_wrap(~order, scales='free_y')+
    theme(legend.position="none",
          text = element_text(size = 20, color = "black"),
          axis.text.x = element_text( vjust = 0.5, hjust=0.5,color = "black", size =20),
          axis.text.y = element_text(color = "black", size =20),
          axis.title.x = element_text(vjust=-1, size = 30, color = "black"),
          axis.title.y  = element_text(color = "black",size = 30),
          panel.border = element_blank(), 
          panel.grid.major = element_blank(),
          panel.grid.minor = element_blank(), 
          axis.line = element_line(colour = "black", size = 1),
          axis.ticks = element_line(colour = "black", size = 1),
          panel.background = element_blank(),
          strip.background = element_blank(),
          strip.text = element_text(size = 30, colour = "black")) +
  scale_x_discrete(breaks=c("17.01","18.01", "19.01", 
                            "20.01", "21.01", "22.01"),
        labels=c("13","14", "15", "16", 
                 "17", "18"))  +
  scale_y_continuous(limits = c(0, NA),labels = formatter1000) +
  ylab("Peptide Abundance (x10^6)") + 
  xlab("June") 

Fig_corr <- total_plot_fig

# manual export
#Fig3_classabundance_red-blue_1000x500_subset
```

Calculate Spearman's correlation coefficient rho for class level summed peptide abundance
```{r}
# wrangle data
taxa_class_corr <- taxa_class_bp
rownames(taxa_class_corr) <- taxa_class_corr$taxon
taxa_class_corr <- taxa_class_corr[, c(2:37)]
colnames(taxa_class_corr) <- substr(colnames(taxa_class_corr), 10, 14)

taxa_class_corr_t <- as.data.frame(t(taxa_class_corr)) # transpose data

# read in metadata to evaluate against ranked time
HAB_binned <- read.csv("C:/Users/mudge/Downloads/HAB_2021_probe_timeseries_binned.csv", header = T, na.strings = T)
HAB_binned$Date.time <- mdy_hm(HAB_binned$Date.hour) #make Date.hour positxct


taxa_class_corr_dt <- cbind(HAB_binned, taxa_class_corr_t) #bind time series and metadata
#taxa_dlm_log_cor_df <- taxa_dlm_log_cor_df[,-1] #remove unnecessary col

taxa_class_corr_dt$Date.hour <- as.numeric(taxa_class_corr_dt$Date.time) #make Date.time numeric so can run "cor" 

# calculate spearman's rho
calculate_correlations_class <- function(df1) {
 # correlations <- list()
  p_values <- numeric(29) # Initialize a vector to store p-values
  rhos <- numeric(29) # Initialize a vector to store rho values
  for (i in 9:37) {
    cor_result <- cor.test(df1[, i], as.numeric(df1[, c("Date.time")]), method = "spearman")
    #correlations[[i]] <- cor_result
    p_values[i] <- cor_result$p.value  # Extract and store the p-value
    rhos[i] <- cor_result$estimate  # Extract and store the rho (correlation coefficient)
  }
  return(data.frame(Peptide = colnames(df1), 
              p_values = p_values, rhos = rhos))
}

# Assuming you have two data frames df_bloom_validation_t1 and df_bloom2_t1
class_corr <- calculate_correlations_class(taxa_class_corr_dt)

# create table
class_corr_table <- as.tibble(class_corr[c(9:37),])

# order correlation table by absolute rho value
class_corr_table <- class_corr_table[rev(order(abs(class_corr_table$rhos))),]

class_corr_export <- as.data.frame(class_corr_table)


#write.csv(class_corr_export, "D:/HAB/Biomarker_Manuscript/class_corr_table.csv")
# Verrucomicrobiae check
#cor.test(taxa_class_corr_dt$Verrucomicrobiae, as.numeric(taxa_class_corr_dt$Date.time), method = c("spearman"))

```
Verrucomicrobiae Spearman's rank correlation rho

data:  taxa_class_corr_dt$Verrucomicrobiae and as.numeric(taxa_class_corr_dt$Date.time)
S = 778, p-value < 2.2e-16
alternative hypothesis: true rho is not equal to 0
sample estimates:
      rho 
0.8998713


#######################################################################
#######################################################################
#######################################################################

Obtain MetaGOmics annotations
```{r}
df_all2 <- read.csv("D:/HAB/2022_09_19_HAB_DDA_metagenome_corrected/MetaGOmics/MetaGOmics_taxa_peptides_abundances_2.csv", header = TRUE)
df_all2 <- df_all2[-1,-1]

library(stringr)

df_all2_uniq <- as.data.frame(str_split_fixed(df_all2$uniqueID, " - ", 4))
names(df_all2_uniq)[names(df_all2_uniq) == "V1"] <- "taxon"
names(df_all2_uniq)[names(df_all2_uniq) == "V2"] <- "taxon_rank"
names(df_all2_uniq)[names(df_all2_uniq) == "V3"] <- "GOterm"
names(df_all2_uniq)[names(df_all2_uniq) == "V4"] <- "GOtype"

df_all2_combo <- cbind(df_all2, df_all2_uniq)

```

Read in and wrangle peptide data
```{r}
#Read in dfs
#time as columns, peptides as rows

# used dataframe to grab Date.Time column so I didn't need to make manually
#just helps set "time" across data
meta_cov <- read.csv("D:/HAB/2022_09_19_HAB_DDA_metagenome_corrected/MetaGOmics/HAB_metadata_MARSS.csv", header = TRUE, na.strings = TRUE)

#HAB_2021_DDA_metagenome_contam_v1.4.10-1.12.31_TICnormalized_092722_36samples_final_Rcleaned
dfpeps_final_zeroesremoved <- read.csv("D:/HAB/2022_09_19_HAB_DDA_metagenome_corrected/HAB_2021_DDA_metagenome_contam_v1.4.10-1.12.31_TICnormalized_092722_36samples_final_Rcleaned.csv", header = TRUE, na.strings = FALSE)

#make peptide names rownames
rownames(dfpeps_final_zeroesremoved) <- dfpeps_final_zeroesremoved[ , 1] 


#get rid of peptide column, keep only quant values
dfpeps_final_zeroesremoved[1] <- NULL 

#make new dataframe to rename runs later
#pepQuant3 <- pepQuant2

#keep only relevant column names (day/time)
names(dfpeps_final_zeroesremoved) <- substring(names(dfpeps_final_zeroesremoved),2,6) 

# make all values numeric
dfpeps_final_zeroesremoved <- mutate_all(dfpeps_final_zeroesremoved, function(x) as.numeric(as.character(x)))
```


Frequency of HMM state shift by time plot
```{r}
# create df for merging
freq_hmm <- dfpeps_final_zeroesremoved
freq_hmm_1 <- freq_hmm
freq_hmm_1$Peptide <- rownames(freq_hmm)

#grab peptides for Phase1A
names(Phase1A)[names(Phase1A) == "rowname"] <- "Peptide"
biomarker_1A <- merge(Phase1A,freq_hmm_1, by = "Peptide", all.x = TRUE)

oneA <- biomarker_1A #447 peptides
rownames(oneA) <- oneA$Peptide 
oneA <- oneA[,c(3:38)]

#grab peptides for Phase1B
df_corr_classes <- df_all2_combo[df_all2_combo['taxon_rank'] == "class", ]
df_corr_classes <- df_corr_classes[rowSums(is.na(df_corr_classes[c(2:37)])) != 36, ] #remove rows of timepoints with all NA to filter out non-peptide rows
df_corr_classes <- df_corr_classes[,c(1:37,39)]
df_corr_classes_uniq <- unique(df_corr_classes)

biomarker_1B <- subset(df_corr_classes_uniq, taxon %in% c("DF_ Alphaproteobacteria", 
                                                   "DF_ Gammaproteobacteria", 
                                                   "DF_ Flavobacteriia",
                                                   "DF_ Verrucomicrobiae",
                                                   "DF_ Opitutae",
                                                   "DF_ Zetaproteobacteria", 
                                                   "DF_ Acidobacteriia"))

oneB <- biomarker_1B #1858 peptides
rownames(oneB) <- oneB$Peptide 
oneB <- oneB[,c(2:37)]
colnames(oneB) <- substring(colnames(oneB),5,9) # clean rownames

#combine all phase 1 peptides
phase1 <- rbind(oneA, oneB) # 2305 peptides
phase1_uniq <- unique(phase1) #2051 peptides

#write.csv(oneB, "D:/HAB/Biomarker_Manuscript/phase1B_peps.csv")
```

Prepare dataframe for HMM
```{r}
phase1_uniq_t <- as.data.frame(t(phase1_uniq))
phase1_uniq_t[phase1_uniq_t == 0] = NA
phase1_uniq_log <- log(phase1_uniq_t)


placeholder <- phase1_uniq_log # set this to keep code consistent with above

# setting some values for code
phase1_hmm_log_imp <- placeholder
a = as.numeric(ncol(placeholder))
b = a + 1
```

Plot peptide abundances for Discovery bloom (Supplementary Figure 8A)
```{r}
pep_list_final <- read.csv("HAB_PRM_peptides_27_controls_standards.csv", header = T, na.strings = F )

dfpeptides_supp8 <- dfpeps_final_zeroesremoved
#colnames(dfpeptides_supp8) <- substr(colnames(dfpeptides_supp8), 4, 8)

dfpeptides_supp8_2 <- cbind(row.names(dfpeptides_supp8),dfpeptides_supp8)

colnames(dfpeptides_supp8_2)[1] = "Peptide" 

pep_bad <- c( # 20 good by HMM
  "IGDYAGIK",     # prtc
  "SAAGAFGPELSR",
"SSAAPPPPPR",
"TASEFDSAIAQDK",
"GISNEGQNASIK")

dfpeptides_supp8_3 <- merge(pep_list_final, dfpeptides_supp8_2, by = "Peptide", all.x = T)
rownames(dfpeptides_supp8_3) <- dfpeptides_supp8_3$Peptide
dfpeptides_supp8_3 <- dfpeptides_supp8_3[, c(3:38)]

dfpeptides_supp8_4 <- dfpeptides_supp8_3[!(row.names(dfpeptides_supp8_3) %in% pep_bad),]
dfpeptides_supp8_4$Peptide <- rownames(dfpeptides_supp8_4)
dfpeptides_supp8_4 <- dfpeptides_supp8_4[, c(37, 1:36)]

dfpeptides_supp8_5 <- gather(dfpeptides_supp8_4, Time, Abundance, c(2:37), factor_key = F) #############################

dfpeptides_supp8_plot <- ggplot(dfpeptides_supp8_5, aes(Time, Abundance, group=1))
dfpeptides_supp8_plot <- dfpeptides_supp8_plot +
  geom_line(data = dfpeptides_supp8_5, aes(x=Time, y=Abundance, group = Peptide),
            colour = 'black', size = 2) +
  facet_wrap(~Peptide, scales='free_y', ncol = 5)+
  theme_minimal() +
  theme(legend.position="bottom",
        axis.text.x = element_text( vjust = 1, 
                                   hjust=0.5, size =40, colour = "black"),
        axis.text.y = element_text(size =40, colour = "black"),
        axis.title.x = element_text(vjust=-1, size =45, colour = "black"),
        axis.title.y  = element_text(size =45, colour = "black"),
        strip.text = element_text(size =30, colour = "black"),
       # strip.text = element_blank(),
        panel.border = element_blank(), 
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(), 
        axis.line = element_line(colour = "black", size = 3)) +
 scale_x_discrete(breaks=c("17.01","18.01", "19.01", 
                            "20.01", "21.01", "22.01"),
        labels=c("13","14", "15", "16", 
                 "17", "18")) +
  scale_y_continuous(limits = c(0, NA),labels = formatter1000) +
  ggtitle("") +
  labs(x='June', y='Peptide Abundance (x10^6)')

dfpeptides_supp8_plot
# Supplementary_Fig_8a
#3500x2500
```

# OPTIONAL
```{r}
# replaces na with mean of the surrounding values... 
# don't need to do this if no missing values - or could leave df with NA

library(zoo)
replace_na_with_mean <- function(x) {
  na_idx <- is.na(x)
  x[na_idx] <- rollapply(x, 3, mean, align = "right", na.rm=TRUE)[na_idx]
  return(x)
}

phase1_processed <- apply(phase1_hmm_log_imp[,1:a], 2, replace_na_with_mean) 
phase1_processed <- as.data.frame(phase1_processed)

```

Run HMM on Phase 1 peptides
```{r}
# IF NOT RUNNING ABOVE CODE CHUNK- RUN THIS INSTEAD
phase1_processed <- placeholder

# get Date.Time column
phase1_processed_2 <- phase1_processed
phase1_processed_2$Date.Time <- rownames(phase1_processed)

# add Date.Time to peptide dataframe
phase1_processed_1 <- cbind(phase1_processed, phase1_processed_2$Date.Time)

colnames(phase1_processed_1) <- colnames(placeholder)
colnames(phase1_processed_1)[b] = "Date.Time" 


# List of peptides to loop through
peptide_list <- as.list(colnames(placeholder[,1:a])) ################

# Create an empty data frame to store results
result_df <- data.frame(Peptide = character(), MaxState = integer())

library(depmixS4)

# Loop through each peptide running HMM
for (peptide in peptide_list) {
  # Filter the data for the current peptide
  ts_tidy_peptide <- phase1_processed_1 %>%
    pivot_longer(!Date.Time, names_to = "Peptide", values_to = "Ratio") %>%
    filter(Peptide == peptide)
  
  # Fit the model
  # Actual HMM code from depmixS4 package
  set.seed(123)
  fit_ts_peptide <- depmix(Ratio ~ 1,
                         nstates = 2,
                         transition = ~1,
                         family = gaussian(),
                         data = ts_tidy_peptide, 
                         type = "viterbi")
  
  # Allows code to run as loop without stalling on error
  tryCatch( {
  fitmod_peptide <- fit(fit_ts_peptide)
  }, 
  error= function(e){
    cat("Error occurred at iteration", conditionMessage(e), "\n")
  })
  
  # Apply which.max to posterior (more code specific to HMM)
  max_state <- apply(posterior(fitmod_peptide)[c("S1", "S2")], 1, which.max)
  
  # Store results in the data frame
  result_df <- rbind(result_df, data.frame(Peptide = peptide, MaxState = max_state))
}
# end loop

##################################
# Print the results - to look at - if you want
# print(result_df)
##################################

result_df$time <- data.frame(time=rep(1:36, times=a)) # rep match number rows of df or number timepoints

# edit df for plotting
pivoted_hmm <- pivot_wider(result_df, names_from = Peptide, values_from = MaxState)
pivoted_hmm_1 <- cbind(pivoted_hmm, phase1_processed_1[c(1:36),b]) # rep match number rows of df or number timepoints
pivoted_hmm_1 <- pivoted_hmm_1[, -1]
colnames(pivoted_hmm_1) <- colnames(phase1_processed_1)

hmm_1 <- as.data.frame(t(pivoted_hmm_1))
hmm_phase2 <- hmm_1
```

Code to plot all peptides evalutated
- takes a long time to run
```{r}
bloom_discovery_HMM <- gather(pivoted_hmm_1, Peptide, state, c(1:a), factor_key = F) #############################
bloom_discovery_HMM <- as.data.frame(bloom_discovery_HMM)
bloom_discovery_HMM %>%
  ggplot(., aes(as.factor(Date.Time), state, group=Peptide)) + # , color=peptide
  geom_line(aes(), show.legend = F) +
  theme(axis.text.x=element_text(angle = 90, size = 6)) +
  ggtitle("Peptides by HMM") +
  facet_wrap(~Peptide)

```

This is to grab only peptides (or features) with non-reverting state shift
aka only 1 change in state
```{r}
# make df numeric
hmm_2 <- mutate_all(hmm_phase2, function(x) as.numeric(as.character(x)))

rownames(hmm_2) <- colnames(pivoted_hmm_1)
colnames(hmm_2) <- rownames(placeholder[1:36,]) # match number rows

# sum number of changes of state across the rows
hmm_2$n_changes <- apply(hmm_2 , MARGIN = 1 , function(x) sum(abs(diff(x))))

# select only peptides with non-reverting state shift
hmm_3 <- hmm_2[hmm_2['n_changes'] == "1", ]

# make column to assign peptide for plotting using rowname
hmm_4 <- hmm_3
hmm_4$Peptide <- rownames(hmm_4)
hmm_4 <- as.data.frame(hmm_4)

#write.csv(hmm_4, "D:/HAB/Biomarker_Manuscript/phase2_peps.csv")
```

Plot underlying state of all non-reverting peptides in discovery data set
```{r}
#plot only nonreverting peptides
bloom_validation_HMM <- gather(hmm_4, Time, State, c(1:36), factor_key = F) #############################
bloom_validation_HMM <- as.data.frame(bloom_validation_HMM)
bloom_validation_HMM %>%
  ggplot(., aes(as.factor(Time), State, group=Peptide)) +
  geom_line(aes(), show.legend = F) +
  theme(axis.text.x=element_text(angle = 90, size = 6)) +
  ggtitle("Bloom 2 Peptides HMM") +
  labs(x='Time', y='State') +
  facet_wrap(~Peptide)
```

Filter for final peptide set
Supplementary Figure 8 b
```{r}
supp_8_ss <- merge(pep_list_final, hmm_4, by = "Peptide", all.x = T)

pep_bad <- c( # 20 good by HMM
  "IGDYAGIK",     # prtc
  "SAAGAFGPELSR",
"SSAAPPPPPR",
"TASEFDSAIAQDK",
"GISNEGQNASIK")

rownames(supp_8_ss) <- supp_8_ss$Peptide
supp_8_ss <- supp_8_ss[!(row.names(supp_8_ss) %in% pep_bad),]


#plot only nonreverting peptides
supp_8_b <- gather(supp_8_ss, Time, State, c(3:38), factor_key = F) #############################

supp_8_b <- as.data.frame(supp_8_b)
supp_8_b %>%
  ggplot(., aes(as.factor(Time), State, group=Peptide)) +
  geom_line(aes(), show.legend = F) +
  theme(axis.text.x=element_text(angle = 90, size = 6)) +
 # ggtitle("Bloom 2 Peptides HMM") +
  labs(x='Time', y='State') +
  facet_wrap(~Peptide)
```

dataframe where rows are ordered by timing for each column
```{r}
# Create a sample dataframe
hmm_5 <- hmm_4[,c(38,1:36)]

hmm_6 <- hmm_5 %>%
  mutate(Peptide = factor(Peptide, levels = hmm_5[,1])) %>%
  arrange(hmm_5[,2:37])

hmm_7 <- hmm_6
hmm_7 <- hmm_7[,-1]
```

create column with index of state shift
```{r}
hmm_7 <- hmm_6
hmm_7 <- hmm_7[,-1]

# Function to identify the index of change within a row
identify_change_index <- function(row) {
  change_index <- which(row[-1] != row[-length(row)])
  if (length(change_index) > 0) {
    return(change_index[1] + 1)  # Adjust index for 1-based indexing
  } else {
    return(NA)
  }
}

# Apply the function to each row of the dataframe
hmm_7$SS <- apply(hmm_7, 1, identify_change_index)

```

make column with timepoint for state shift 
```{r}
hmm_8 <- hmm_7

# Function to print colname(df[i]) as new column in df
print_colname_as_new_column <- function(col_values, df) {
  new_column <- colnames(df)[col_values]
  return(new_column)
}

# Apply the function to create new column
hmm_8$timepoint <- print_colname_as_new_column(hmm_8$SS, hmm_8)  # Example: index 2
#write.csv(hmm_8, "D:/HAB/Biomarker_Manuscript/biomarker_phase2_peptides.csv")
```

Generate frequency plot
```{r}

met.sig20 <- met.brewer("Signac", 20) 

# Create a bar plot of the frequency of each value in the column
freq_plot <- ggplot(hmm_8, aes(x = timepoint)) +
          geom_bar(fill =met.sig20[2], color = met.sig20[1] ) +
          labs(x = "June", y = "Number of Peptides") + #,title="Frequency of All Non-reverting Peptides with State Shift by Time") +
    theme_bw() + 
  theme(legend.position="none",
          text = element_text(size = 20, color = "black"),
          axis.text.x = element_text(angle = 0, vjust = 0.5, hjust=0.5,color = "black", size =20),
          axis.text.y = element_text(color = "black", size =20),
          axis.title.x = element_text(vjust=-1, size = 30, color = "black"),
          axis.title.y  = element_text(color = "black",size = 30),
          panel.border = element_blank(), 
          panel.grid.major = element_blank(),
          panel.grid.minor = element_blank(), 
          axis.line = element_line(colour = "black", size = 1),
          axis.ticks = element_line(colour = "black", size = 1)) +
  scale_x_discrete(breaks=c("17.01","18.01", "19.01", 
                            "20.01", "21.01", "22.01"),
                   labels=c("13","14", "15", 
                            "16", "17", "18"), 
                   expand = c(0,0)) 
Fig_freq <- freq_plot
#Fig3C_new_800x800
```

### code to differentiate, of peptides that undergo non-reverting state shift, which are increasing vs decreasing in abundance and when 
- not included in paper

To make df of all information from Phase 2 for ~300 potential biomarker peptides
```{r}
hmm_9 <- hmm_8
hmm_9$Peptide <- rownames(hmm_9)
hmm_9 <- hmm_9[,c(39,38,37)]

phase3 <- merge(hmm_9, phase1_uniq, by = 0, all.x = TRUE)
```

Find average before and after SS
```{r}
phase3_foraverage <- phase3[,c(5:40,4)]
rownames(phase3_foraverage) <- phase3$Peptide


# Function to calculate row means before and after the index column
calculate_averages <- function(row) {
  # Get the index value
  index_value <- row["SS"]
  
  # Find the index column
#  index_col <- which(names(row) == "SS")
  
  # Calculate the row means before and after the index column
  mean_before <- mean(row[1:(index_value - 1)], na.rm = TRUE)
  mean_after <- mean(row[(index_value ):36], na.rm = TRUE)
  
  # Return the calculated means
  return(c(mean_before, mean_after))
}


# Apply the function to each row of the dataframe
averages <- t(apply(phase3_foraverage, 1, calculate_averages))

# Create a new dataframe with the averages
averages_df <- data.frame(Avg_Before = averages[, 1], Avg_After = averages[, 2])

#test
#before <- rowMeans(phase3_foraverage[1, 1:16], na.rm = TRUE)
#after <- rowMeans(phase3_foraverage[1, 17:36], na.rm = TRUE)


#find if peptide abundance is increasing or decreasing
phase3$trend <- ifelse(phase3[, 5] > phase3[, 40], "down", "up")

# Compare the averages and fill the "abundance" column
phase3$trend_2 <- ifelse(averages_df$Avg_Before > averages_df$Avg_After, "down", "up")
# note that 23 peptides have discrepancy between first v last value and average before v after ss
```

To separate number of peptides undergoing state shift into those peptides with abundances increasing vs decreasing
-based on average abundance prior to state shift
```{r}
phase3_down <- phase3[phase3['trend_2'] == "down", ]
phase3_up <- phase3[phase3['trend_2'] == "up", ]


# Create a bar plot of the frequency of each value in the column
freq_plot_down <- ggplot(phase3_down, aes(x = timepoint)) +
          geom_bar() +
          labs(x = "Date", y = "Frequency", title = "Frequency of Decreasing Peptides with State Shift by Time") +
    theme_bw() + 
  theme(legend.position="none",
          text = element_text(size = 10, color = "black"),
          axis.text.x = element_text(angle = 0, vjust = 0.5, hjust=0.5,color = "black", size =15),
          axis.text.y = element_text(color = "black", size =15),
          axis.title.x = element_text(vjust=-1, size = 18, color = "black"),
          axis.title.y  = element_text(color = "black",size = 18),
          panel.border = element_blank(), 
          panel.grid.major = element_blank(),
          panel.grid.minor = element_blank(), 
          axis.line = element_line(colour = "black", size = 1),
          axis.ticks = element_line(colour = "black", size = 1)) +
  scale_x_discrete(breaks=c("17.01","18.01", "19.01", 
                            "20.01", "21.01", "22.01"),
                   labels=c("13","14", "15", 
                            "16", "17", "18")) 


# Create a bar plot of the frequency of each value in the column
freq_plot_up <- ggplot(phase3_up, aes(x = timepoint)) +
          geom_bar() +
          labs(x = "Date", y = "Frequency", title = "Frequency of Increasing Peptides with State Shift by Time") +
    theme_bw() + 
  theme(legend.position="none",
          text = element_text(size = 10, color = "black"),
          axis.text.x = element_text(angle = 0, vjust = 0.5, hjust=0.5,color = "black", size =15),
          axis.text.y = element_text(color = "black", size =15),
          axis.title.x = element_text(vjust=-1, size = 18, color = "black"),
          axis.title.y  = element_text(color = "black",size = 18),
          panel.border = element_blank(), 
          panel.grid.major = element_blank(),
          panel.grid.minor = element_blank(), 
          axis.line = element_line(colour = "black", size = 1),
          axis.ticks = element_line(colour = "black", size = 1)) +
  scale_x_discrete(breaks=c("17.01","18.01", "19.01", 
                            "20.01", "21.01", "22.01"),
                   labels=c("13","14", "15", 
                            "16", "17", "18")) 
```

To plot all peptide abundances for peptides that undergo a state shift
```{r}
phase3_forplot <- phase3[,c(2,5:40)]

#plot only nonreverting peptides
bloom_discovery_HMM <- gather(phase3_forplot, Time, State, c(2:37), factor_key = F) #############################
bloom_discovery_HMM <- as.data.frame(bloom_discovery_HMM)
bloom_discovery_HMM %>%
  ggplot(., aes(as.factor(Time), State, group=Peptide)) +
  geom_line(aes(), show.legend = F, size = 2, color = "black") +
  theme_minimal() +
  theme(axis.text.x = element_text( vjust = 1, 
                                   hjust=1, size =30, colour = "black"),
        axis.text.y = element_text(size =30, colour = "black"),
        axis.title.x = element_text(vjust=-1, size =35, colour = "black"),
        axis.title.y  = element_text(size =35, colour = "black"),
        strip.text = element_text(size =30, colour = "black"),
        panel.border = element_blank(), 
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(), 
        axis.line = element_line(colour = "black", size = 3)) +
  ggtitle("Disovery Bloom Peptide Abundances for peptides with non-reverting state shift prior to BSD via HMM") +
   scale_x_discrete(breaks=c("17.01","18.01", "19.01", 
                            "20.01", "21.01", "22.01"),
        labels=c("13","14", "15", "16", 
                 "17", "18")) +
  scale_y_continuous(limits = c(0, NA),labels = formatter1000) +
    labs(x='June', y='Peptide Abundance (x 10^6') +
  facet_wrap(~Peptide, scales='free_y' )

```

#########################################################################################################################
#########################################################################################################################
#########################################################################################################################

# Code for calculation and figures related to Figure 4


Take PRM set of 27 peptides plus controls and standards and append to discovery data set peptide abundance values
```{r}
#df_PRM <- read.csv("D:/HAB/PRM_104_peptides.csv", header = TRUE)
df_PRM <- read.csv("D:/HAB/Biomarker_Manuscript/HAB_PRM_peptides_27_controls_standards.csv", header=T, na.strings = F)
df_PRM <- as.data.frame(df_PRM[,2])
names(df_PRM)[names(df_PRM) == "df_PRM[, 2]"] <- "Peptide"


PRM_27 <- merge(df_PRM, df_all2_combo, by = "Peptide", all.x = T)

PRM_27_uniq <- PRM_27[,1:37]
PRM_27_uniq_1 <- unique(PRM_27_uniq)
PRM_27_uniq <- PRM_27_uniq_1[,-1]
rownames(PRM_27_uniq) <- PRM_27_uniq_1$Peptide
names(PRM_27_uniq) <- substring(names(PRM_27_uniq),5,9) #get rid of prefix to taxon, do this here not later
PRM_27_uniq <- PRM_27_uniq[rowSums(is.na(PRM_27_uniq[c(1:36)])) != 36, ] #remove rows of timepoints with all NA to filter out non-peptide rows


PRM_27_uniq <- PRM_27_uniq[c( 
  "LAGEIFDASEGR",
"IPAQADGTPER",
"SAFLSSDPTSPSR",
"SSSGIANQAVK"
),]


#Make df for HMM code bloom 2

pep_PRM_t <- as.data.frame(t(PRM_27_uniq))
pep_PRM_t1 <- as.data.frame(lapply(pep_PRM_t, as.numeric))
pep_PRM_log <- log(pep_PRM_t1)
pep_PRM_log[pep_PRM_log == "-Inf"] <- NA  #all negative values = NA
pep_PRM_log[pep_PRM_log == 0] <- NA  #all zero values = NA
rownames(pep_PRM_log) <- rownames(pep_PRM_t)



#HMM Discovery Bloom

placeholder <- pep_PRM_log
a = as.numeric(ncol(placeholder))
b = a + 1
d = as.numeric(nrow(pep_PRM_log))


PRM_fig_processed_1 <- pep_PRM_log
PRM_fig_processed_1$Date.Time <- rownames(pep_PRM_log)

# List of peptides to loop through
peptide_list_fig4_disovery <- as.list(colnames(placeholder[,1:a])) ################
# Create an empty data frame to store results
result_df_fig4_discovery <- data.frame(Peptide = character(), MaxState = integer())

library(depmixS4)

# Loop through each class
for (peptide in peptide_list_fig4_disovery) {
  # Filter the data for the current peptide
  ts_tidy_peptide <- PRM_fig_processed_1 %>%
    pivot_longer(!Date.Time, names_to = "Peptide", values_to = "Abundance") %>%
    filter(Peptide == peptide)
  
  # Fit the model
  set.seed(123)
  fit_ts_peptide <- depmix(Abundance ~ 1,
                         nstates = 2,
                         transition = ~1,
                         family = gaussian(),
                         data = ts_tidy_peptide)
  
  tryCatch( {
  fitmod_peptide <- fit(fit_ts_peptide)

  }, 
  error= function(e){
    cat("Error occurred at iteration", conditionMessage(e), "\n")
  })

  
  # Apply which.max to posterior
  max_state <- apply(posterior(fitmod_peptide)[c("S1", "S2")], 1, which.max)
  
  
  # Store results in the data frame
  result_df_fig4_discovery <- rbind(result_df_fig4_discovery, data.frame(Peptide = peptide, MaxState = max_state))
}

# Print the results
print(result_df_fig4_discovery)
result_df_fig4_discovery$time <- data.frame(time=rep(1:d, times=a)) ##################################
pivoted_df_fig4_discovery <- pivot_wider(result_df_fig4_discovery, names_from = Peptide, values_from = MaxState)


pivoted_df_fig4_discovery_1 <- pivoted_df_fig4_discovery[, -1]
rownames(pivoted_df_fig4_discovery_1) <- rownames(pep_PRM_log)
pivoted_df_fig4_discovery_1$Date.Time <- rownames(pivoted_df_fig4_discovery_1)
#colnames(pivoted_df_1) <- colnames(pep_PRM_log)

df_fig4_discovery <- as.data.frame(t(pivoted_df_fig4_discovery_1))
```


Set start state to 1
```{r}
# Function to swap values 1 and 2 in each row of a dataframe, excluding "Date.Time" column
swap_values_in_rows <- function(df) {
  # Exclude "Date.Time" column
  df_without_date <- df[row.names(df) != "Date.Time", ]
  
  # Loop through each row
  for (i in 1:nrow(df_without_date)) {
    # Check if value in the first column of the row is "2"
    if (df_without_date[i, 1] == "2") {
      # Swap values "1" and "2" in the row
      df_without_date[i, -1] <- ifelse(df_without_date[i, -1] == "1", "2", "1")

    }
  }
  
  # Re-combine with the "Date.Time" column
  df_result <- rbind(df[rownames(df) == "Date.Time", , drop = FALSE], df_without_date)
  
  return(df_result)
}

df_swap <- swap_values_in_rows(df_fig4_discovery)
df_swap[, 1][-1] <- "1"

#move first row to last
df_fig4_discovery <- rbind(df_swap[-1, ], df_swap[1, ])

# move first column to be last
df_fig4_discovery_test <- as.data.frame(t(df_swap))
pivoted_df_fig4_discovery_1 <- df_fig4_discovery_test[, c(2:ncol(df_fig4_discovery_test), 1)]

#pivoted_df <- pivoted_df_1[,2:30]
Discovery_HMM_fig4 <- gather(pivoted_df_fig4_discovery_1, peptide, state, c(1:a), factor_key = F) 
```


```{r}
#State shift for bloom 2 plot
# this is where to get timing of the state shift
library(MetBrewer)
met.vg7 <-met.brewer("VanGogh3", 7) #scale green
met.vg7[5]


###################################
###################################
###################################

fig4_discovery_2 <- mutate_all(df_fig4_discovery, function(x) as.numeric(as.character(x)))

rownames(fig4_discovery_2) <- colnames(pivoted_df_fig4_discovery_1)
colnames(fig4_discovery_2) <- rownames(placeholder[1:d,])

fig4_discovery_2$n_changes <- apply(fig4_discovery_2 , MARGIN = 1 , function(x) sum(abs(diff(x))))

fig4_discovery_3 <- fig4_discovery_2[fig4_discovery_2['n_changes'] == "1", ]

fig4_discovery_3$Peptide <- rownames(fig4_discovery_3)


###################################
###################################
# could make code to select df 24 hrs prior to bloom, set n_changes to zero, 
# merge peptides back to HMM df and plot only these peptides

fig4_discovery_6 <- mutate_all(df_fig4_discovery, function(x) as.numeric(as.character(x)))

rownames(fig4_discovery_6) <- colnames(pivoted_df_fig4_discovery_1)
colnames(fig4_discovery_6) <- rownames(placeholder[1:d,])


# select 24 or 48 hrs prior to bloom
#fig4_discovery_7 <- fig4_discovery_6[,c(24:36)]
fig4_discovery_7 <- fig4_discovery_6
####################################
fig4_discovery_7$n_changes <- apply(fig4_discovery_7 , MARGIN = 1 , function(x) sum(abs(diff(x))))
fig4_discovery_8 <- fig4_discovery_7[fig4_discovery_7['n_changes'] == "1", ]

fig4_discovery_8$Peptide <- rownames(fig4_discovery_8)
fig4_discovery_9 <- fig4_discovery_8[,c(38,1:37)]

bloom_discovery_ss <- gather(fig4_discovery_9, Sample, State, c(2:37), factor_key=TRUE)

```

Plot abundances of PRM initial set only
```{r}
PRM_final <- merge(fig4_discovery_9, PRM_27_uniq_1, by = "Peptide", all.x = T) #merge set of prm peps from above by HMM only 1 ss with train bloom abundance values

PRM_discovery_values <- PRM_final[, c(39:74)] #grab only abundance values
rownames(PRM_discovery_values) <- PRM_final$Peptide #keep labels
names(PRM_discovery_values) <- substring(names(PRM_discovery_values),5,9) #get rid of prefix to taxon, do this here not later

PRM_discovery <- cbind(rownames(PRM_discovery_values), PRM_discovery_values) #give Peptide column for plotting
names(PRM_discovery)[names(PRM_discovery) == "rownames(PRM_discovery_values)"] <- "Peptide" #change col name
```

Biomarker peptide abundances Discovery bloom
```{r Fig 4A}
##################################################

df_peptide_total_4 <- gather(PRM_discovery, Sample, Abundance, c(2:37), factor_key=TRUE)

df_peptide_total_4$Abundance[is.na(df_peptide_total_4$Abundance)] <- 0

df_bloom_discovery_abu_ss <- merge(df_peptide_total_4, bloom_discovery_ss, by = c("Peptide", "Sample"), all = T)
df_bloom_discovery_abu_ss$State<-as.character(df_bloom_discovery_abu_ss$State)
```

Figure 4A
```{r Fig 4A}
################3

# Set colors based on the levels of 'new'
colors_ss <- c(met.sig20[16],met.sig20[19] ) 


df_peptide_total_plot_b <- ggplot(df_bloom_discovery_abu_ss, aes(Sample, Abundance, group=1))
df_peptide_total_plot_b <- df_peptide_total_plot_b +
  geom_line(data = df_peptide_total_4, aes(x=Sample, y=Abundance, group = Peptide),
            colour = 'black', size = 2) +
  facet_wrap(~Peptide, scales='free_y', ncol = 5)+
  theme_minimal() +
  theme(legend.position="bottom",
        axis.text.x = element_text( vjust = 1, 
                                   hjust=0.5, size =40, colour = "black"),
        axis.text.y = element_text(size =40, colour = "black"),
        axis.title.x = element_text(vjust=-1, size =45, colour = "black"),
        axis.title.y  = element_text(size =45, colour = "black"),
      #  strip.text = element_text(size =30, colour = "black"),
        strip.text = element_blank(),
        panel.border = element_blank(), 
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(), 
        axis.line = element_line(colour = "black", size = 3)) +
 scale_x_discrete(breaks=c("17.01","18.01", "19.01", 
                            "20.01", "21.01", "22.01"),
        labels=c("13","14", "15", "16", 
                 "17", "18")) +
  scale_y_continuous(limits = c(0, NA),labels = formatter1000) +
  ggtitle("") +
  labs(x='June', y='Peptide Abundance (x10^6)') 
#df_peptide_total_plot_A

Fig4_a <- df_peptide_total_plot_b
Fig4_a

#ggsave(file="D:/HAB/test_Fig3A.png", plot=Fig3_A, width=30, height=15, bg = "white")
#supp_discovery_abundance_2500x1500
```
####################################################################################################
####################################################################################################
####################################################################################################

Matrix Matched Calibration Curve All
```{r}
#read in data
pep_30 <- read.csv("D:/HAB/2024_01_08_HAB_PRM_test/HAB_PRM_bloom1_quant_results_MMCC.csv", header=T, na.strings = F)

#clean dataframe
MMCC <- pep_30[,c(2:8)]
rownames(MMCC) <- MMCC$Peptide


#reorder samples
MMCC <- MMCC[,c(1, 7, 6, 5, 4, 3, 2)]

#make tidy for plotting
MMCC_tidy <- gather(MMCC, Sample, Abundance, c(2:7), factor_key=TRUE)

########################
# Create a new column 'Concentration' based on the 'Sample' column
MMCC_tidy$Concentration <- factor(MMCC_tidy$Sample, levels = c(
  "PRM5_F_30.Normalized.Area",
  "PRM5_E_29.Normalized.Area",
  "PRM5_D_28.Normalized.Area",
  "PRM5_C_27.Normalized.Area",
  "PRM5_B_26.Normalized.Area",
  "PRM5_A_25.Normalized.Area"
), labels = c("0:100", "0.3:99.7", "3:97", "50:50", "70:30", "100:0"))

MMCC_tidy$Concentration_numeric <- as.numeric(gsub(":.*", "", MMCC_tidy$Concentration))

MMCC_tidy$Group1 <- ifelse(MMCC_tidy$Concentration_numeric %in% c(0, 0.3), "Group1", "Other")
MMCC_tidy$Group2 <- ifelse(MMCC_tidy$Concentration_numeric %in% c(0.3, 3, 50, 70, 100), "Group2", "Other")

met.sig14 <- met.brewer("Signac", 14) 

# plot
pep_30_MMCC <- ggplot(MMCC_tidy, aes(Concentration_numeric, Abundance, group = Peptide))
pep_30_MMCC <- pep_30_MMCC +
  geom_point(aes(), show.legend = FALSE, size = 10) +
    scale_x_continuous(breaks = c(0,  50,  100), labels = c("0:1",  "0.5:0.5", "1:0")) +
  theme_minimal() +
  facet_wrap(~Peptide, scales = 'free_y', ncol = 5) + #, nrow = 1) +
  theme(legend.position="none",
        axis.text.x = element_text( vjust = 1, hjust=0.5, size =60, colour = "black"),
        axis.text.y = element_text(size =60, colour = "black"),
        axis.title.x = element_text(vjust=-1, size =65, colour = "black"),
        axis.title.y  = element_text(size =65, colour = "black"),
        strip.text = element_text(size = 60),
        panel.border = element_blank(), 
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(), 
        axis.line = element_line(colour = "black", size = 2.5)) +
    scale_y_continuous(limits = c(0, NA),labels = formatter1000) +
  ylab("Peptide Abundance (x10^6)") +
  xlab("Ratio of Microbiome to Matched Matrix")
pep_30_MMCC

Supp_Fig_7 <- pep_30_MMCC
#6500x3000
#Supplementary_Fig_7_mod.jpeg
```

####################################################################################################
####################################################################################################
####################################################################################################

Fig 4 Validation Bloom analysis
```{r}
pep_30_S <- read.csv("D:/HAB/2024_01_08_HAB_PRM_test/HAB_PRM_bloom1_quant_results_samples_all.csv", header=T, na.strings = F)

pep_samples <- pep_30_S[,c(2:39)]


pep_prm <- pep_samples[,c(2:38)]
rownames(pep_prm) <- pep_samples$Peptide


pep_PRM_good <- pep_prm
#pep_PRM_good <- pep_PRM_good[,-1]
names(pep_PRM_good) <- sub("X", "", names(pep_PRM_good))
names(pep_PRM_good) <- substring(names(pep_PRM_good), 1, 4)
names(pep_PRM_good) <- sub("X", "", names(pep_PRM_good))
#names(pep_PRM_good)[names(pep_PRM_good) == "21"] <- "0021" 
#names(pep_PRM_good)[names(pep_PRM_good) == "701"] <- "0701"

pep_PRM_good_1 <- cbind(rownames(pep_PRM_good), pep_PRM_good)

names(pep_PRM_good_1)[names(pep_PRM_good_1) == "rownames(pep_PRM_good)"] <- "Peptide" 

pep_PRM_good_1 <- pep_PRM_good_1[c( 
  "LAGEIFDASEGR",
"IPAQADGTPER",
"SAFLSSDPTSPSR",
"SSSGIANQAVK"
),]

pep_C_tidy <- gather(pep_PRM_good_1, Sample, Abundance, c(2:38), factor_key=TRUE)
```

```{r}

plot_fig4b <- ggplot(pep_C_tidy, aes(Sample, Abundance, group=Peptide))
plot_fig4b <- plot_fig4b +
  geom_line(data = pep_C_tidy, aes(x=Sample, y=Abundance, group = Peptide),
            colour = 'black', size = 2) +
 # scale_color_manual(values = colors_ss) +  # Set manual colors for geom_smooth
  #scale_fill_manual(values = colors_ss) +
  facet_wrap(~Peptide, scales='free_y', ncol = 5)+
  theme_minimal() +
  theme(legend.position="none",
        axis.text.x = element_text( vjust = 1, 
                                   hjust=0.5, size =40, colour = "black"),
        axis.text.y = element_text(size =40, colour = "black"),
        axis.title.x = element_text(vjust=-1, size =45, colour = "black"),
        axis.title.y  = element_text(size =45, colour = "black"),
        strip.text = element_text(size =40, colour = "black"),
       # strip.text = element_blank(),
        panel.border = element_blank(), 
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(), 
        axis.line = element_line(colour = "black", size = 3)) +
  scale_x_discrete(breaks=c("0101","0201", "0301", 
                            "0401", "0501", "0601"),
        labels=c("28","29", "30", "31", 
                 "01", "02")) +
    scale_y_continuous(limits = c(0, NA),labels = formatter1000) +
  ggtitle("") +
  ylab("Peptide Abundance (x10^6)") + 
  xlab("May - June") 


Fig4_b <- plot_fig4b

```

Fig 4 Validation Bloom analysis All PRM candidate peptides
```{r}
pep_30_S <- read.csv("D:/HAB/2024_01_08_HAB_PRM_test/HAB_PRM_bloom1_quant_results_samples_all.csv", header=T, na.strings = F)

pep_samples <- pep_30_S[,c(2:39)]


pep_prm <- pep_samples[,c(2:38)]
rownames(pep_prm) <- pep_samples$Peptide


pep_PRM_good <- pep_prm
#pep_PRM_good <- pep_PRM_good[,-1]
names(pep_PRM_good) <- sub("X", "", names(pep_PRM_good))
names(pep_PRM_good) <- substring(names(pep_PRM_good), 1, 4)
names(pep_PRM_good) <- sub("X", "", names(pep_PRM_good))
#names(pep_PRM_good)[names(pep_PRM_good) == "21"] <- "0021" 
#names(pep_PRM_good)[names(pep_PRM_good) == "701"] <- "0701"

pep_PRM_good_1 <- cbind(rownames(pep_PRM_good), pep_PRM_good)

names(pep_PRM_good_1)[names(pep_PRM_good_1) == "rownames(pep_PRM_good)"] <- "Peptide" 

pep_C_tidy <- gather(pep_PRM_good_1, Sample, Abundance, c(2:38), factor_key=TRUE)
```

```{r}

supp_fig_9 <- ggplot(pep_C_tidy, aes(Sample, Abundance, group=Peptide))
supp_fig_9 <- supp_fig_9 +
  geom_line(data = pep_C_tidy, aes(x=Sample, y=Abundance, group = Peptide),
            colour = 'black', size = 2) +
 # scale_color_manual(values = colors_ss) +  # Set manual colors for geom_smooth
  #scale_fill_manual(values = colors_ss) +
  facet_wrap(~Peptide, scales='free_y', ncol = 5)+
  theme_minimal() +
  theme(legend.position="none",
        axis.text.x = element_text( vjust = 1, 
                                   hjust=0.5, size =40, colour = "black"),
        axis.text.y = element_text(size =40, colour = "black"),
        axis.title.x = element_text(vjust=-1, size =45, colour = "black"),
        axis.title.y  = element_text(size =45, colour = "black"),
        strip.text = element_text(size =40, colour = "black"),
       # strip.text = element_blank(),
        panel.border = element_blank(), 
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(), 
        axis.line = element_line(colour = "black", size = 3)) +
  scale_x_discrete(breaks=c("0101","0201", "0301", 
                            "0401", "0501", "0601"),
        labels=c("28","29", "30", "31", 
                 "01", "02")) +
    scale_y_continuous(limits = c(0, NA),labels = formatter1000) +
  ggtitle("") +
  ylab("Peptide Abundance (x10^6)") + 
  xlab("May - June") 


supp_fig_9_a <- supp_fig_9


# code to save individual file
#ggsave(file="D:/HAB/Biomarker_Manuscript/Fig3_test_peps_4.png", plot=pep_C_plot, width=30, height=12)

#Supp_Fig_9_a
#3500x2500

```

## Code for validation bloom
```{r}

peps_prm <- read.csv("D:/HAB/peps_prm.csv", header = T, na.strings = F)
peps_prm <- peps_prm[,-1]

df_peptide <- peps_prm[,c(1:27,29:33,35:38)] # remove bad ms runs

df_peptide_num <- as.data.frame(lapply(df_peptide[2:36],as.numeric)) #make pep abundances numeric based on col number
names(df_peptide_num) <- substring(names(df_peptide_num),2,5) #get rid of prefix to taxon, do this here not later

df_peptide_combo <- cbind(df_peptide$Peptide, df_peptide_num) #bind numeric df back into OG df
names(df_peptide_combo)[names(df_peptide_combo) == "df_peptide$Peptide"] <- "Peptide" #change col name

df_peptide_red <- unique(df_peptide_combo) #remove redundant entries


pep_PRM_test <- df_peptide_red
pep_PRM_test2 <- pep_PRM_test[,-1]
rownames(pep_PRM_test2) <- pep_PRM_test$Peptide 

pep_PRM_t <- as.data.frame(t(pep_PRM_test2))
pep_PRM_t1 <- as.data.frame(lapply(pep_PRM_t, as.numeric))
pep_PRM_log <- log(pep_PRM_t1)
pep_PRM_log[pep_PRM_log == "-Inf"] <- NA  #all negative values = 0
pep_PRM_log[pep_PRM_log == 0] <- NA  #all negative values = 0
rownames(pep_PRM_log) <- rownames(pep_PRM_t)
pep_PRM_log <- pep_PRM_log[c(1:34),]
```

HMM validation bloom
```{r}
placeholder <- pep_PRM_log
a = as.numeric(ncol(placeholder))
b = a + 1
d = as.numeric(nrow(pep_PRM_log))


phase1_processed_1 <- pep_PRM_log
phase1_processed_1$Date.Time <- rownames(pep_PRM_log)

# List of peptides to loop through
peptide_list <- as.list(colnames(placeholder[,1:a])) ################
# Create an empty data frame to store results
result_df <- data.frame(Peptide = character(), MaxState = integer())

library(depmixS4)

# Loop through each peptide
for (peptide in peptide_list) {
  # Filter the data for the current peptide
  ts_tidy_peptide <- phase1_processed_1 %>%
    pivot_longer(!Date.Time, names_to = "Peptide", values_to = "Abundance") %>%
    filter(Peptide == peptide)
  
  
  # Fit the model
  set.seed(123)
  fit_ts_peptide <- depmix(Abundance ~ 1,
                         nstates = 2,
                         transition = ~1,
                         family = gaussian(),
                         data = ts_tidy_peptide)
  
  tryCatch( {
  fitmod_peptide <- fit(fit_ts_peptide)
  }, 
  error= function(e){
    cat("Error occurred at iteration", conditionMessage(e), "\n")
  })
  
  # Apply which.max to posterior
  max_state <- apply(posterior(fitmod_peptide)[c("S1", "S2")], 1, which.max)
  
  # Store results in the data frame
  result_df <- rbind(result_df, data.frame(Peptide = peptide, MaxState = max_state))
}

# Print the results
print(result_df)

result_df$time <- data.frame(time=rep(1:d, times=a)) ##################################
pivoted_df <- pivot_wider(result_df, names_from = Peptide, values_from = MaxState)


pivoted_df_1 <- pivoted_df[, -1]
rownames(pivoted_df_1) <- rownames(pep_PRM_log)
pivoted_df_1$Date.Time <- rownames(pivoted_df_1)
#colnames(pivoted_df_1) <- colnames(pep_PRM_log)

df_1 <- as.data.frame(t(pivoted_df_1))
colnames(df_1) <- t(pivoted_df_1$Date.Time)
```


```{r}
df_swap_b1 <- swap_values_in_rows(df_1)
df_swap_b1[, 1][-1] <- "1"

#move first row to last
df_1 <- rbind(df_swap_b1[-1, ], df_swap_b1[1, ])

df_ss_1 <- df_1
df_ss_1$Peptide <- rownames(df_ss_1)
df_ss_1 <- df_ss_1[,c(35,1:34)]


# make data tidy for plotting
bloom_validation_ss <- gather(df_ss_1, Sample, State, c(2:35), factor_key=TRUE)


df_bloom_validation_abu_ss <- merge(pep_C_tidy, bloom_validation_ss, by = c("Peptide", "Sample"), all.x = T)
df_bloom_validation_abu_ss$State<-as.character(df_bloom_validation_abu_ss$State)

df_bloom_validation_abu_ss <- df_bloom_validation_abu_ss[df_bloom_validation_abu_ss['State'] != "0505", ] #poor quality sample
df_bloom_validation_abu_ss <- df_bloom_validation_abu_ss[df_bloom_validation_abu_ss['State'] != "0605", ] #poor quality sample
df_bloom_validation_abu_ss <- df_bloom_validation_abu_ss[df_bloom_validation_abu_ss['State'] != "0701", ] #outside range of investigation

df_bloom_validation_abu_ss <- df_bloom_validation_abu_ss[complete.cases(df_bloom_validation_abu_ss), ]
```

HMM of Validation experiment of 4 representative peptides
```{r}
pivoted_df_1 <- pivoted_df_1[,c( 
  "LAGEIFDASEGR",
"IPAQADGTPER",
"SAFLSSDPTSPSR",
"SSSGIANQAVK",
"Date.Time"
)]

#pivoted_df <- pivoted_df_1[,2:30]
#bloom_validation_HMM <- gather(pivoted_df_1, peptide, state, c(1:d), factor_key = F) #############################
met.vg7 <-met.brewer("VanGogh3", 7) #scale green
met.vg7[5]

bloom_validation_HMM <- gather(pivoted_df_1, Peptide, State, c(1:4), factor_key = F) #############################
bloom_validation_HMM <- as.data.frame(bloom_validation_HMM)

#fig3_test_HMM_34 <- bloom_validation_HMM %>%
 # ggplot(., aes(as.factor(Date.Time), state, group=Peptide)) 
  
fig4_validation_HMM_4 <- ggplot(bloom_validation_HMM, aes(Date.Time, State, group=Peptide))
fig4_validation_HMM_4 <- fig4_validation_HMM_4 +
  geom_line(data = bloom_validation_HMM, aes(x=Date.Time, y=State, group = Peptide),
            colour = 'black', size = 2) +  geom_vline(xintercept = "0501", 
             linetype="dotted", color = met.vg7[7], size = 2) +
  theme_minimal() +
  theme(legend.position="none",
        axis.text.x = element_text( vjust = 1, 
                                   hjust=0.5, size =40, colour = "black"),
        axis.text.y = element_text(size =40, colour = "black"),
        axis.title.x = element_text(vjust=-1, size =45, colour = "black"),
        axis.title.y  = element_text(size =45, colour = "black"),
        strip.text = element_text(size =40, colour = "black"),
        panel.border = element_blank(), 
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(), 
        axis.line = element_line(colour = "black", size = 3)) +
    scale_x_discrete(breaks=c("0101","0201", "0301", 
                            "0401", "0501", "0601", "0602"),
        labels=c("28","29", "30", "31", 
                 "01", "02", "03")) +
  ggtitle("") +
  labs(x='May - June', y='State') +
  facet_wrap(~Peptide, scales='free_y', ncol = 5)

Supp_Fig_b <- fig4_validation_HMM_4

#ggsave(file="D:/HAB/Fig3_test_HMM_34.svg", plot=fig3_test_HMM_34, width=30, height=15)
#supp_validation_ss_2500x1500
```


##################################################################################

HMM of Validation experiment all peptides
Plot Supplementary Figure 9 B
```{r}
met.vg7 <-met.brewer("VanGogh3", 7) #scale green
met.vg7[5]

Supp_9_b <- ggplot(df_bloom_validation_abu_ss, aes(Sample, State, group=Peptide))
Supp_9_b <- Supp_9_b +
  geom_line(data = df_bloom_validation_abu_ss, aes(x=Sample, y=State, group = Peptide),
            colour = 'black', size = 2) +  geom_vline(xintercept = "0501", 
             linetype="dotted", color = met.vg7[7], size = 2) +
  theme_minimal() +
  theme(legend.position="none",
        axis.text.x = element_text( vjust = 1, 
                                   hjust=0.5, size =40, colour = "black"),
        axis.text.y = element_text(size =40, colour = "black"),
        axis.title.x = element_text(vjust=-1, size =45, colour = "black"),
        axis.title.y  = element_text(size =45, colour = "black"),
        strip.text = element_text(size =40, colour = "black"),
        panel.border = element_blank(), 
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(), 
        axis.line = element_line(colour = "black", size = 3)) +
    scale_x_discrete(breaks=c("0101","0201", "0301", 
                            "0401", "0501", "0601", "0602"),
        labels=c("28","29", "30", "31", 
                 "01", "02", "03")) +
  ggtitle("") +
  labs(x='May - June', y='State') +
  facet_wrap(~Peptide, scales='free_y', ncol = 5)

Supp_Fig_9_b <- Supp_9_b
Supp_Fig_9_b

#Supp_Fig_9_b
#3500x2500
```

Plot all Fig 4 panels
```{r}

library(cowplot)
################3
#plot_grid(Fig3_A, Fig3_B, Fig3_C, Fig3_D, Fig3_E, align = "v", nrow = 5, rel_heights = c(1/3, 1/8, 1/4, 1/3, 1/8))
plot_grid(Fig4_a,  Fig4_b,  align = "v", nrow = 3, rel_heights = c(1, 1))

#Fig4_final_2500x3000.png
```


###############################################################################
###############################################################################
###############################################################################

#Supplemental figures Extra


Filter by taxa, include annotations for candidate peptides
```{r}

df_class <- df_all2_combo[df_all2_combo['taxon_rank'] == "no rank", ] #select only class from taxon_rank

##### grab peptides from PRM analysis
peps_prm <- merge(pep_PRM_good_1, df_all2_combo[,c(1:37)], by = "Peptide", all.x = T)

df_class <- peps_prm[,c(1,39:74)] 

df_class_num <- as.data.frame(lapply(df_class[2:37],as.numeric)) #make pep abundances numeric based on col number
names(df_class_num) <- substring(names(df_class_num),5,9) #get rid of prefix to taxon, do this here not later

df_class_combo <- cbind(df_class$Peptide, df_class_num) #bind numeric df back into OG df
names(df_class_combo)[names(df_class_combo) == "df_class$Peptide"] <- "Peptide" #change col name

df_class_red <- df_class_combo[rowSums(is.na(df_class_combo[c(2:37)])) != 36, ] #remove rows of timepoints with all NA to filter out non-peptide rows

df_class_red <- unique(df_class_combo)
#write.csv(df_corr_classes_uniq, "D:/HAB/Biomarker_Manuscript/Supplementary_Data_5.csv")
```

Align timing to BSD
```{r}
df_bloom_validation <- pep_PRM_good_1[,c(1:25)] #24 timepoints
df_bloom2 <- df_class_red[,c(1,14:37)] 

# remove external controls for plotting
df_bloom_validation <- df_bloom_validation[!(row.names(df_bloom_validation) %in% pep_bad),]
df_bloom2 <- df_bloom2[!(row.names(df_bloom2) %in% pep_bad),]

generic_times <- 0:24

colnames(df_bloom_validation) <- generic_times
colnames(df_bloom2) <- generic_times

names(df_bloom_validation)[names(df_bloom_validation) == "0"] <- "Peptide" 
names(df_bloom2)[names(df_bloom2) == "0"] <- "Peptide" 

df_bloom_validation_tidy <- gather(df_bloom_validation, Sample, Time, c(2:25), factor_key=TRUE)
df_bloom2_tidy <- gather(df_bloom2, Sample, Time, c(2:25), factor_key=TRUE)

df_bloom_merge <- merge(df_bloom_validation_tidy, df_bloom2_tidy, by=c( "Peptide", "Sample"))

df_bloom_merge$Sample <- lapply(df_bloom_merge$Sample,as.numeric)

order = c("1","2","3","4","5","6","7","8","9",
          "10","11","12","13","14","15","16","17","18","19",
          "20","21","22","23","24","25")

#df_bloom_merge$order <- factor(df_bloom_merge$order, levels = c(1:34))
met.sig20 <- met.brewer("Signac", 20) 

Supplementary_Fig_10 <- ggplot(df_bloom_merge, aes(Time.x, Time.y)) + 
  geom_point(mapping = aes(x = Time.x, y = Time.y), size = 10) +
    geom_point(mapping = aes(x = Time.y, y = Time.x), alpha = 0) +
  geom_abline(slope = 1, linetype = "dashed", color = "black") +
  facet_wrap(~Peptide, scales='free') + 
  xlab("Validation dataset peptide abundance (x10^6)") + 
  ylab("Discovery dataset peptide abundance (x10^6)") + 
  #ggtitle("State shift timing aligned to BSD") +
  theme_bw() + 
      theme(legend.position="none",
        axis.text.x = element_text( vjust = 1, 
                                   hjust=0.5, size =40, colour = "black"),
        axis.text.y = element_text(size =40, colour = "black"),
        axis.title.x = element_text(vjust=-1, size =50, colour = "black"),
        axis.title.y  = element_text(size =50, colour = "black"),
        strip.text = element_text(size =40, colour = "black"),
        panel.border = element_blank(), 
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(), 
        axis.line = element_line(colour = "black", size = 3),
          axis.ticks = element_line(colour = "black", size = 1),
          panel.background = element_blank(),
          strip.background = element_blank()) +
  geom_smooth(aes(group=1, fill = met.sig20[16]),method="lm", se=T, color =met.sig20[16] ) +
    scale_fill_manual(values = met.sig20[16]) +
      scale_y_continuous(limits = c(0, NA),labels = formatter1000) +
    scale_x_continuous(limits = c(0, NA),labels = formatter1000) 



# Supplementary_Fig_10
# 4000x4000
```

Spearman's correlation coefficient rho for Supplementary Table 3
```{r}
# transpose data for spearmans corr
df_bloom_validation_t <- df_bloom_validation
df_bloom2_t <- df_bloom2
rownames(df_bloom2_t) <- df_bloom2_t$Peptide

df_bloom2_t <- df_bloom2_t[!(row.names(df_bloom2_t) %in% pep_bad),]

df_bloom_validation_t <- df_bloom_validation_t[,-1]
df_bloom2_t <- df_bloom2_t[,-1]


df_bloom_validation_t1 <- as.data.frame(t(df_bloom_validation_t))
df_bloom_validation_t1 <- df_bloom_validation_t1[,order(colnames(df_bloom_validation_t1))]
df_bloom_validation_t1 <- as.matrix(df_bloom_validation_t1)

df_bloom2_t1 <- t(df_bloom2_t)

calculate_correlations <- function(df1, df2) {
 # correlations <- list()
  p_values <- numeric(29) # Initialize a vector to store p-values
  rhos <- numeric(29) # Initialize a vector to store rho values
  for (i in 1:29) {
    cor_result <- cor.test(df1[, i], df2[, i], method = "spearman")
    #correlations[[i]] <- cor_result
    p_values[i] <- cor_result$p.value  # Extract and store the p-value
    rhos[i] <- cor_result$estimate  # Extract and store the rho (correlation coefficient)
  }
  return(data.frame(Peptide = colnames(df1), 
              p_values = p_values, rhos = rhos))
}

# Assuming you have two data frames df_bloom_validation_t1 and df_bloom2_t1
results <- calculate_correlations(df_bloom_validation_t1, df_bloom2_t1)

class_corr_table_all <- results[rev(order(abs(results$rhos))),]

class_corr_export_all <- as.data.frame(class_corr_table_all)

#write.csv(class_corr_export_all, "D:/HAB/Biomarker_Manuscript/Supplementary_Dataset_7.csv")

```


