# If package installation required, run the following code
```{r}
############################################################################
############################################################################

## Install from CRAN

# install.packages("tidyverse") # insert name of package into " " 

############################################################################
############################################################################
```


```{r}
#packages needed

library(lubridate)
library(ggplot2)
library(tidyr)
library(MetBrewer)
library(tidyverse)
library(dplyr)
library(scales)
library(cowplot)
library(reshape2)

#files needed: 
 
  # Nunn_2021_light_metadata.csv
  # Nunn_2021_YSI_probe_data.csv 
  # Nunn_2021_flowcytodata.csv
  # Nunn_2021_128_metagenome_class_counts.csv
  # Nunn_2021_128_JGI_ID_Timepoints.csv
  # Nunn_2021_Nuts_metadata.csv
  # hr.pts.csv

```


```{r Set color Pallet}
#scale_color_metbrewer(palette = "Signac")
met.sig14 <- met.brewer("Signac", 14) #14 colors
met.rdn<-met.brewer("Redon",12 )



```




## Plotting corrected Light values over 22 days
```{r Plotting corrected Light values over 22 days}

#set color pallet
met.sig14 <- met.brewer("Signac", 14) #14 colors

#load light corrected masterfile that has times when the probe was removed mid-day for downloads
masterfile <- read.csv("Nunn_2021_light_metadata.csv", header = TRUE, na.strings = TRUE)

#set Date ID
masterfile$DateID <- as.POSIXct(masterfile$DateTime, format = "%m/%d/%Y %H:%M")  
names(masterfile)[names(masterfile) == "Intensity....lum.ft2."] <- "Light"

#select numeric columns to plot and remove Date
master_numeric <- masterfile[,c(2:21,24:27,29:31,33)]

#make values to numeric
master_numeric2 <- mutate_all(master_numeric, function(x) as.numeric(as.character(x)))

#column bind together Date ID back to these new columns with numbers
master_numeric3 <- cbind(masterfile$DateID, master_numeric2)

names(master_numeric3)[names(master_numeric3) == "masterfile$DateID"] <- "DateID"


#Generate line plot that is filled in 
Light.full <- ggplot(master_numeric3, aes(DateID, Light)) +
  geom_area(fill = met.sig14[1], alpha = 1) +  # Plot area first
  geom_line(color = met.sig14[1], size = 0.2, alpha = 1) +  # Plot line next
  ylim(0, 1000) +
 
  labs(
    y = expression(bold("Light (luminosity/ft"^"2"~")")),
    title = "Light"
  ) +
  theme_bw() +
  scale_y_continuous(expand=c(0,0))+
  scale_x_datetime(labels = date_format("%d"), breaks = "24 hours",
                  limits = c(as.POSIXct("2021-05-27 01:00:00"), 
                           max(as.POSIXct("2021-06-18 13:00:00"))),
                  expand=c(0,0)
                   ) +
  theme(
    text = element_text(size = 10, face = "bold"),
    axis.text.x = element_text(angle = 0, vjust = 0.5, hjust = 0.5),
    axis.text = element_text(color = "black"),
    axis.title.y = element_text(face = "bold"),
    panel.border = element_blank(),
    panel.grid.major = element_blank(),
    panel.grid.minor = element_blank(),
    axis.line = element_line(colour = "black", size = 1),
    axis.ticks = element_line(colour = "black", size = 1),
    plot.margin = margin(0, 0, 0,0, "cm")  # Adjust the margins
  )

Light.full

```


```{r Save plot light data}
#Plot light data

# Light.full  is ggplot object to save
ggsave("Light_full.png", Light.full, width = 6, height = 2, dpi = 300)

#save as 1000 x 600 as svg
#################

```



## prepping Probe data
```{r prepping prope data for chl plots}
# getting data ready for the next R chuck plot
combined_ts_2 <- read.csv("Nunn_2021_YSI_probe_data.csv", header = T, na.strings = F)

#set color pallet
met.sig14 <- met.brewer("Signac", 14) #14 colors

# one way to make date into actual date
combined_ts_2 <- combined_ts_2 %>%
  mutate(Date = mdy(Date)) 


# make Date.Time into correct class
combined_ts_2$Date.Time = ymd_hms(paste(combined_ts_2$Date, combined_ts_2$Time))

combined_ts_2$hour <- format(strptime(combined_ts_2$Time, "%H:%M"), "%H:00")

#Create a Date + Time column
combined_ts_2$Date.hour <- with(combined_ts_2, ymd(Date) + hm(hour)) #create column to group date.time by hour

#combines data across time to match field sample timepoints
combined_bin1 <- combined_ts_2 %>% group_by(Date.hour) %>% 
  summarize(Chlorophyll.ugL = mean(Chlorophyll.ugL))    #group chl to match hour bins 

combined_bin2 <- combined_ts_2 %>% group_by(Date.hour) %>% 
  summarize(Temp.C = mean(Temp.C)) 

combined_bin3 <- combined_ts_2 %>% group_by(Date.hour) %>% 
  summarize(Conductivity.uscm = mean(Conductivity.uscm)) 

combined_bin4 <- combined_ts_2 %>% group_by(Date.hour) %>% 
  summarize(Sal.psu = mean(Sal.psu))

combined_bin5 <- combined_ts_2 %>% group_by(Date.hour) %>% 
  summarize(pH = mean(pH))

combined_bin6 <- combined_ts_2 %>% group_by(Date.hour) %>% 
  summarize(ODO.sat = mean(ODO.sat))

combined_bin <- merge(combined_bin1,combined_bin2,by="Date.hour")
combined_bin <- merge(combined_bin,combined_bin3,by="Date.hour")
combined_bin <- merge(combined_bin,combined_bin4,by="Date.hour")
combined_bin <- merge(combined_bin,combined_bin5,by="Date.hour")
combined_bin <- merge(combined_bin,combined_bin6,by="Date.hour")

combined_bin$Chlorophyll.ugL[combined_bin$Chlorophyll.ugL<0] <- 0

combined_chl <- combined_bin
names(combined_chl)[names(combined_chl) == "Date.hour"] <- "DateID"

met.vg7 <-met.brewer("VanGogh3", 7) #scale green

```


Generate line plot with Stdev of bact flow cytometry data.
```{r Bacterial Flow cytometry data}
library(ggplot2)

# Read the CSV file
bact.flow <- read.csv("Nunn_2021_flowcytodata.csv", header = TRUE, na.strings = FALSE)

#set color pallet
met.sig14 <- met.brewer("Signac", 14) #14 colors

# Convert DateID to POSIXct format
bact.flow$DateIDnew <- as.POSIXct(bact.flow$DateID, format = "%m/%d/%y %H:%M")

# Assign variables for standard deviation and average
sd <- bact.flow$std.bact
avg <- bact.flow$avg.bact

# Create the plot
bacteria.flow.plot <- ggplot(bact.flow, aes(x = DateIDnew, y = avg)) + 
  geom_point(aes(x = DateIDnew, y = bact.rep1), color = met.sig14[8], size = 0.8) +
    geom_point(aes(x = DateIDnew, y = bact.rep2), color = met.sig14[8], size = 0.8) +
  geom_line(color = met.sig14[6]) +
 # geom_ribbon(aes(ymin = avg - sd, ymax = avg + sd), fill = met.sig14[6], alpha = 0.6) +
  labs(x = "Date", y = "Bacterial counts", title = "Bacterial Flow Cytometry Counts") +
    scale_y_continuous(limits = c(0,1.2E7), expand=c(0,0))+
    scale_x_datetime(labels = date_format("%d"), breaks = "24 hours",
                  limits = c(as.POSIXct("2021-05-27 01:00:00"),
                           max(as.POSIXct("2021-06-18 13:00:00"))),
                  expand=c(0,0)
                   ) + 
  #ylim(0, 15000000) +
  theme_bw() +
  theme(
    text = element_text(size = 10, face = "bold"),
    axis.text.x = element_text(angle = 0, vjust = 0.5, hjust = 0.5),
    axis.text = element_text(color = "black"),
    axis.title.y = element_text(face = "bold"),
    panel.border = element_blank(),
    panel.grid.major = element_blank(),
    panel.grid.minor = element_blank(),
    axis.line = element_line(colour = "black", size = 1),
    axis.ticks = element_line(colour = "black", size = 1)
  )
bacteria.flow.plot
# Set the x-axis limits to align with the desired start date
#bacteria.flow.plot + xlim(as.POSIXct("2021-05-27 13:00"), max(bact.flow$DateIDnew))



# Light.full  is ggplot object to save
ggsave("D:/SciData paper/bact_flow_full.png", bacteria.flow.plot, width = 6.4, height = 2, dpi = 300)
```





Generate line plot with Stdev of pico, nano, cyano flow cytometry data.
```{r Eukarote Flow cytometry data}
library(ggplot2)

# Read the CSV file
euk.flow <- read.csv("Nunn_2021_flowcytodata.csv", header = TRUE, na.strings = FALSE)

#set color pallet
met.sig14 <- met.brewer("Signac", 14) #14 colors

# Convert DateID to POSIXct format
euk.flow$DateIDnew <- as.POSIXct(euk.flow$DateID, format = "%m/%d/%y %H:%M")


#########Pico plot############

# Assign variables for standard deviation and average
pico.sd <- euk.flow$std.pico
pico.avg <- euk.flow$avg.pico

# Create the plot
pico.flow.plot <- ggplot(euk.flow, aes(x = DateIDnew, y = pico.avg)) + 
  geom_line(color = met.sig14[11]) +
  geom_ribbon(aes(ymin = pico.avg - pico.sd, ymax = pico.avg + pico.sd), fill = met.sig14[10], alpha = 0.5) +
  labs(x = "Date", y = "Pico counts", title = "Pico Flow Cytometry Counts") +
    scale_y_continuous(expand=c(0,0))+
    scale_x_datetime(labels = date_format("%d"), breaks = "24 hours",
                  limits = c(as.POSIXct("2021-05-27 01:00:00"),
                           max(as.POSIXct("2021-06-18 13:00:00"))),
                  expand=c(0,0)
                   ) + 
 
  theme_bw() +
  theme(
    text = element_text(size = 10, face = "bold"),
    axis.text.x = element_text(angle = 0, vjust = 0.5, hjust = 0.5),
    axis.text = element_text(color = "black"),
    axis.title.y = element_text(face = "bold"),
    panel.border = element_blank(),
    panel.grid.major = element_blank(),
    panel.grid.minor = element_blank(),
    axis.line = element_line(colour = "black", size = 1),
    axis.ticks = element_line(colour = "black", size = 1)
  )
pico.flow.plot
 



#  is ggplot object to save
ggsave("pico_flow_full.png", pico.flow.plot, width = 6.4, height = 2, dpi = 300)

#########Nano plot############

# Assign variables for standard deviation and average
nano.sd <- euk.flow$std.nano
nano.avg <- euk.flow$avg.nano

# Create the plot
nano.flow.plot <- ggplot(euk.flow, aes(x = DateIDnew, y = nano.avg)) + 
  geom_line(color = met.sig14[3]) +
  geom_ribbon(aes(ymin = nano.avg - nano.sd, ymax = nano.avg + nano.sd), fill = met.sig14[3], alpha = 0.5) +
  labs(x = "Date", y = "Nano counts", title = "Nano Flow Cytometry Counts") +
    scale_y_continuous(expand=c(0,0))+
    scale_x_datetime(labels = date_format("%d"), breaks = "24 hours",
                  limits = c(as.POSIXct("2021-05-27 01:00:00"),
                           max(as.POSIXct("2021-06-18 13:00:00"))),
                  expand=c(0,0)
                   ) + 
 
  theme_bw() +
  theme(
    text = element_text(size = 10, face = "bold"),
    axis.text.x = element_text(angle = 0, vjust = 0.5, hjust = 0.5),
    axis.text = element_text(color = "black"),
    axis.title.y = element_text(face = "bold"),
    panel.border = element_blank(),
    panel.grid.major = element_blank(),
    panel.grid.minor = element_blank(),
    axis.line = element_line(colour = "black", size = 1),
    axis.ticks = element_line(colour = "black", size = 1)
  )
nano.flow.plot
 



#  is ggplot object to save
ggsave("nano_flow_full.png", nano.flow.plot, width = 6.4, height = 2, dpi = 300)

#########Cyano plot############

# Assign variables for standard deviation and average
cyano.sd <- euk.flow$std.cyano
cyano.avg <- euk.flow$avg.cyano

# Create the plot
cyano.flow.plot <- ggplot(euk.flow, aes(x = DateIDnew, y = cyano.avg)) + 
  geom_line(color = met.sig14[13]) +
  geom_ribbon(aes(ymin = cyano.avg - cyano.sd, ymax = cyano.avg + cyano.sd), fill = met.sig14[13], alpha = 0.5) +
  labs(x = "Date", y = "cyano counts", title = "cyano Flow Cytometry Counts") +
    #scale_y_continuous(expand=c(0,0))+
    ylim(0,25000)+
    scale_x_datetime(labels = date_format("%d"), breaks = "24 hours",
                  limits = c(as.POSIXct("2021-05-27 01:00:00"),
                           max(as.POSIXct("2021-06-18 13:00:00"))),
                  expand=c(0,0)
                   ) + 
 
  theme_bw() +
  theme(
    text = element_text(size = 10, face = "bold"),
    axis.text.x = element_text(angle = 0, vjust = 0.5, hjust = 0.5),
    axis.text = element_text(color = "black"),
    axis.title.y = element_text(face = "bold"),
    panel.border = element_blank(),
    panel.grid.major = element_blank(),
    panel.grid.minor = element_blank(),
    axis.line = element_line(colour = "black", size = 1),
    axis.ticks = element_line(colour = "black", size = 1)
  )
cyano.flow.plot
 



#  is ggplot object to save
ggsave("cyano_flow_full.png", cyano.flow.plot, width = 6.4, height = 2, dpi = 300)
```


```{r Generate plot of nutrients through time}
library(ggplot2)
library(lubridate)
library(MetBrewer)

cov.dat<-read.csv('Nunn_2021_Nuts_metadata.csv')
cov.dat$DateTime2<-paste(cov.dat$Date, cov.dat$Time)
cov.dat$DateTime2<-parse_date_time(cov.dat$DateTime2, "mdy HM")

cov.nut<-subset(cov.dat, select=c('DateTime2', "NH4", 'PO4', 'Si.OH.4', 'NO3', 'NO2'))


met.sig14 <- met.brewer('Signac', 14) #14 colors

nuts.plot <- ggplot(cov.nut) +
    geom_line(aes(x=DateTime2, y=NH4), color=met.sig14[10], size=.8) +   #purple
     geom_line(aes(x=DateTime2, y=PO4), color=met.sig14[8], size=.8)+     #pink
    geom_line(aes(x=DateTime2, y=Si.OH.4), color=met.sig14[13], size=.8)+  #teal green
    geom_line(aes(x=DateTime2, y=NO3), color=met.sig14[12], size=1)+       #coblt blue
    geom_line(aes(x=DateTime2, y=NO2), color=met.sig14[14], size=.8)+      #lime green
        labs(x = "Date", y = "Nutrient Concentrations", title = "Nutrients") +
          #scale_y_continuous(expand=c(0,0))+
        scale_x_datetime(labels = date_format("%d"), breaks = "24 hours",
                  limits = c(as.POSIXct("2021-05-27 01:00:00"),
                           max(as.POSIXct("2021-06-18 13:00:00"))),
                  expand=c(0,0)
                   ) +
  theme_bw() +
  theme(
    text = element_text(size = 10, face = "bold"),
    axis.text.x = element_text(angle = 0, vjust = 0.5, hjust = 0.5),
    axis.text = element_text(color = "black"),
    axis.title.y = element_text(face = "bold"),
    panel.border = element_blank(),
    panel.grid.major = element_blank(),
    panel.grid.minor = element_blank(),
    axis.line = element_line(colour = "black", size = 1),
    axis.ticks = element_line(colour = "black", size = 1)
  )
nuts.plot

#  is ggplot object to save
ggsave("nutrients.png", nuts.plot, width = 6.4, height = 2, dpi = 300)

```




BLN Generate one plot with grey actual data and green averaged on the hour data Chl.a
```{r Generate one plot with grey actual data and green averaged on the hour data Chl.a}

##### Generate one plot with grey actual data and green averaged on the hour data Chl.a
#load csv that has Date.hour for actual sample times
hour.pts<- read.csv("hr.pts.csv", header = T, na.strings = F)
hour.pts <- hour.pts %>%
  mutate(Date = mdy(Date)) 

hour.pts$Time <- format(strptime(hour.pts$hour, "%H:%M"), "%H:00")

# make Date.Time into correct class
hour.pts$Date.Time = ymd_hms(paste(hour.pts$Date, hour.pts$hour))


#Create a Date + Time column
hour.pts$Date.hour <- with(hour.pts, ymd(Date) + hm(hour))

chl.plot<- ggplot()+

  geom_line(data=combined_ts_2, aes(x=Date.hour, y=Chlorophyll.ugL), color='grey', alpha=0.5) +
  geom_area(data=combined_bin, aes(x=Date.hour, y=Chlorophyll.ugL), fill=met.sig14[14], alpha=0.8) +
  geom_line(data=combined_bin, aes(x=Date.hour, y=Chlorophyll.ugL), color=met.sig14[13], alpha = 1)+
   theme_bw()+
  geom_point(data=hour.pts, aes(x=Date.Time, y=-5), color= met.sig14[10], size = .8, shape=17)+
 ylim(-5, 250)+
     #scale_y_continuous(expand=c(0,0))+
    scale_x_datetime(labels = date_format("%d"), breaks = "24 hours",
                  limits = c(as.POSIXct("2021-05-27 01:00:00"),
                           max(as.POSIXct("2021-06-18 13:00:00"))),
                  expand=c(0,0)
                   ) +

  theme(text = element_text(size = 10, face="bold"),
    axis.text.x = element_text(angle = 0, vjust = 0.5, hjust=0.5),
        axis.text = element_text(color = "black"),
    axis.title.y  = element_text(face="bold"),
        panel.border = element_blank(), 
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(), 
        axis.line = element_line(colour = "black", size = 1),
    axis.ticks = element_line(colour = "black", size = 1)) 
chl.plot
 
```
```{r Save plot chl data}
#Plot chl data

# Light.full  is ggplot object to save
ggsave("Chl_full.png", chl.plot, width = 6, height = 4, dpi = 300)


```

Making Taxonomy plots (Taz Nunn)
``` {r }
library(readxl)
library(dplyr)
library(tidyverse)
library(reshape2)
library(dplyr)

# Read files
JGI_taxa <- read.csv("Nunn_2021_128_metagenome_class_counts.csv")
JGI_ID <- read.csv("Nunn_2021_128_JGI_ID_Timepoints.csv")

#make JGI_ID_Raw$Timepoint into recognied month/day/year 00:00 format
JGI_ID$TimepointDate <- mdy_hm(JGI_ID$Timepoint)

# Rename column
colnames(JGI_taxa)[1] <- "taxon_oid"


# Function to calculate percentages for each row
ClassPercent_128 <- function(df) {
  # Select numeric columns 
  class_cols <- colnames(df)[-which(colnames(df) == "taxon_oid")]
  df_class_cols <- df[, class_cols]
 # View(df_class_cols) #includes only classes
  
  # Loop through each row start with row 1
  for (i in 1:nrow(df_class_cols)) {
    # Calculate the sum of each row- 
    row_sum <- sum(df_class_cols[i,  ], na.rm = TRUE)
    
    # Divide each element in the row by the sum to get the percentage
    df[i, class_cols ] <- df_class_cols[i,  ] / row_sum * 100
     
  }
  
  return(df)
}


# Apply the function to the merged data frame
Taxonomy_128_PCT <- ClassPercent_128(JGI_taxa)

# Merging
Taxonomy.percents_IDs_128 <- merge(Taxonomy_128_PCT, JGI_ID, by = "taxon_oid", all = TRUE)

# Define main taxa columns to exclude
main_taxa <- c("Alphaproteobacteria", "Gammaproteobacteria", "Flavobacteriia", "Betaproteobacteria", "Verrucomicrobiae", "Cytophagia", "Bacteroidia", "Actinomycetes", "Cyanophyceae", "Opitutae", "Bacilli", "other", "unclassified", "TimepointDate", "taxon_oid", "Timepoint")
main_taxa_cols <- which(names(Taxonomy.percents_IDs_128) %in% main_taxa)

# Create a new column with combined values from all columns except main_taxa
Taxonomy.percents_IDs_128 <- Taxonomy.percents_IDs_128 %>%
  mutate(OtherClasses = rowSums(Taxonomy.percents_IDs_128[, -main_taxa_cols], na.rm = TRUE))





FinalTaxaPCT <- subset(Taxonomy.percents_IDs_128, select = c(Alphaproteobacteria, Gammaproteobacteria, Flavobacteriia, Cytophagia, Verrucomicrobiae, Betaproteobacteria, unclassified, Bacteroidia, Actinomycetes, Cyanophyceae, Opitutae, Saprospiria, OtherClasses, TimepointDate))


#Generate the stacked line plot
#scale_color_metbrewer(palette = "Signac")
met.sig14 <- rev(met.brewer("Signac", 14)) #14 colors rev=reverses the order


#FINAL PLOT YAY
FinalTaxaPCT_Melt <- melt(FinalTaxaPCT, id.vars = "TimepointDate") 

taxa.dist <-ggplot(FinalTaxaPCT_Melt, aes(x = TimepointDate, y = value,color = variable, group = variable, fill = variable)) +
  scale_color_manual(values = met.sig14) +
  scale_fill_manual(values = met.sig14) +
  geom_area(alpha = .85) +
  labs(x = "Timepoint", y = "Relative gene count", title = "Classes") +
  #theme_minimal()
     scale_y_continuous(expand=c(0,0))+
  scale_x_datetime(labels = date_format("%d"), breaks = "24 hours",
                  limits = c(as.POSIXct("2021-05-27 01:00:00"),
                           max(as.POSIXct("2021-06-18 13:00:00"))),
                  expand=c(0,0)
                   ) +
  
   theme_bw()+
  theme(text = element_text(size = 10, face="bold"),
    axis.text.x = element_text(angle = 0, vjust = 0.5, hjust=0.5),
        axis.text = element_text(color = "black"),
    axis.title.y  = element_text(face="bold"),
        panel.border = element_blank(), 
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(), 
        axis.line = element_line(colour = "black", size = 1),
    axis.ticks = element_line(colour = "black", size = 1),
    legend.position = "bottom",  # Position legend at the bottom
    legend.box = "horizontal" ) 
    
print(taxa.dist)

```
```{r Save plot taxa data}
#Plot taxa data

# Taxa.dist  is ggplot object to save
ggsave("taxa_distrib.png", taxa.dist, width = 6, height = 5, dpi = 300)


```
Names to plot (top 14)
Alphaproteobacteria
Gammaproteobacteria
Flavobacteriia
Betaproteobacteria
Verrucomicrobiae
Cytophagia
unclassified
Bacteroidia
Actinomycetes
Cyanophyceae
Opitutae
Bacilli
other
unclassified

good order
Alphaproteobacteria, Gammaproteobacteria, Flavobacteriia, Cytophagia, Verrucomicrobiae, Betaproteobacteria, unclassified, Bacteroidia, Actinomycetes, Cyanophyceae, Opitutae, Saprospiria, Bacilli, Sphingobacteriia, Clostridia
